
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="robots" content="noindex">
<meta name="ipynb_website:version" content="0.9.4" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=writeupDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=writeupArray;
            var docs_map=writeupArrayMap;
            var pos=writeupArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>GTEx eQTL</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">GTEx eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
<li>
  <a href="../prototype.html">Prototype</a>
</li>
        
<li>
  <a href="../writeup.html">Writeup</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/stephenslab/gtex-eqtls"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Meetings">Meetings<a class="anchor-link" href="#Meetings">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="20180108-GTEx-V8-data-status-update">20180108 GTEx V8 data status update<a class="anchor-link" href="#20180108-GTEx-V8-data-status-update">&#182;</a></h2><p>AWG call.</p>
<p>V8 release timeline: End of December 2017 internal release. End of June 2018 public release (dbGAP embargo period).</p>
<p>Francois, recap of status:</p>
<ul>
<li>End of December 2017: data was largely complete. 838 donors, ~16,000 RNA seq samples</li>
<li>Major change is phased + imputed genotype, analysis are updated<ul>
<li>with exception of: allelic fold change calculation and sQTL results (<code>leafcutter</code>) but both will be available next week</li>
<li>eQTL results has been updated with imputed genotypes</li>
</ul>
</li>
<li>Annotation files etc are uploaded for fine-mapping</li>
</ul>
<p>Tuuli:</p>
<ul>
<li>The need to compile "official" GTEx regulatory annotations from ensembl + encode roadmap</li>
</ul>
<p>Tuuli, V8 paper:</p>
<ul>
<li>Strict reproducibility is perhaps not realistic given the time frame, but we should put all scripts and documentation of analysis in github, for transparency.<ul>
<li>Just a github directory for GTEx V8 main paper, and consolidate people's work</li>
<li>Will be made public when the paper comes out</li>
<li>Problems:<ul>
<li>Manolis: How do you link existing repo to main paper repo? Redudant work for people who already has organized separate github repos.</li>
<li>Francois: How about scripts private to a cluster environment? <ul>
<li>Tuuli: compensate with good documentation; does not have to be runnable.</li>
</ul>
</li>
<li>Manolis: Dumping everything in github but not runnable will likely lead to lots of critism and complaints. Maybe share core scripts is enough.<ul>
<li>Francois and Tuuli disagree that people will attack such practice.</li>
<li>Manolis insists that people expect Consortium work come with quality and completness not sloppiness ... thus deeply worried. Maybe into each individual analysts github repos would be better, not under GTEx umbrella and not intended to be user-friendly. But make the "key" methodology scripts very well documented and runnable, not those one-liners, hacks and data processing.</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li>Companion papers: Google doc will be sent out to the group.<ul>
<li>Any papers should be discussed with AWG; no unpleasant surprises.</li>
<li>Submission embargo? biorxiv embargo? End of 2018 or submission of main paper whichever first.</li>
</ul>
</li>
</ul>
<p>Kristin:</p>
<ul>
<li>Push out main paper; review presentation list</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="20171211-GTEx-fine-mapping-workgroup-meeting">20171211 GTEx fine mapping workgroup meeting<a class="anchor-link" href="#20171211-GTEx-fine-mapping-workgroup-meeting">&#182;</a></h2><p>Phone conference with GTEx members from many cities ... Andrew presented.</p>
<p>Software to apply:</p>
<ul>
<li>Farhad Caviar: PIP on all tissues, 3 days</li>
<li>Andrew Caveman: PIP on all tissues, but whole blood first, a couple of days; all tissues 3 weeks</li>
<li>William DAP-G: a couple of days per tissue. A couple of tissues to start from.<ul>
<li>PIP</li>
<li>LD clusters</li>
</ul>
</li>
<li>Possibly 1 or 2 more software</li>
<li>Results for all tissues will be available mid-Jan.</li>
</ul>
<p>Data:</p>
<ul>
<li>Francois: VCF are almost final, though missingness are causing confusion (but in short we can mean-impute them). Should be finalized mid-week this week</li>
<li>Andrew: ChIP-seq benchmarks (an annotation list) to be shared. The data comes from roadmap epigenomics and will be lifted over to V9 data.</li>
<li>Tuuli: Has annotation for post transcriptional modification</li>
<li>William: Has annotation for TFBS.</li>
</ul>
<p>Simulation data:</p>
<ul>
<li>Andrew: not necessary. or at least should not be too close to the way any methods work. eg, no one-eqtl per gene type of setup.</li>
<li>William: is necessary. no need to worry about generating from certain models. Simulation based on simple linear model is generic enough.</li>
<li>Eleazar: Simulation results not interesting, compared to ChIP-seq / other annotation benchmarks.</li>
</ul>
<p>Exploration then benchmarking, or benchmarking right now?</p>
<ul>
<li>Blood results before Christmas, so that benchmarking can be put together.</li>
</ul>
<p>Benchmarking:</p>
<ul>
<li>Input: data and annotation. Annotation can be used for tests, or for validation<ul>
<li>Caveman now taks pre-computed annotations</li>
<li>Some type of Enrichment summary</li>
</ul>
</li>
<li>Output: scores?<ul>
<li>Gao has to figure out these</li>
<li>PIP, lfdr? at variant vs. gene level? how about LD cluster based methods?</li>
</ul>
</li>
<li>Blood results ready in a week to start putting together benchmarks. </li>
<li>All tissues by mid-Jan<ul>
<li>data availability</li>
</ul>
</li>
</ul>
<p>Extension of methods:</p>
<ul>
<li>cell-specific eQTLs</li>
<li>splicing eQTLs</li>
<li>age dependent QTLs</li>
<li>co-localisation<ul>
<li>Haky: GWAS group is also interested</li>
</ul>
</li>
<li>multiple traits<ul>
<li>Tuuli: can anyone put together a quick&amp;dirty M&amp;M analysis? <ul>
<li>Tuuli's group is developing something</li>
</ul>
</li>
<li>Andrew: pairwise tissue analysis</li>
<li>William: can look at a few tissues</li>
</ul>
</li>
</ul>
<h2 id="20171002-GTEx-fine-mapping-meeting">20171002 GTEx fine mapping meeting<a class="anchor-link" href="#20171002-GTEx-fine-mapping-meeting">&#182;</a></h2><p>Goal: methods do not repeat each-other; should build on each other.</p>
<h3 id="What-is-fine-mapping">What is fine-mapping<a class="anchor-link" href="#What-is-fine-mapping">&#182;</a></h3><ul>
<li>probability estimates of each variant being causal<ul>
<li>how to verify it? experimentally?</li>
</ul>
</li>
<li>Should use 1MB region?<ul>
<li>100KB should not lose a lot.</li>
</ul>
</li>
</ul>
<h3 id="Fine-mapping-methods">Fine-mapping methods<a class="anchor-link" href="#Fine-mapping-methods">&#182;</a></h3><p>Analyse/benchmark one data-set using all methods. The best tissue might be Thyroid to test it out.</p>
<ul>
<li>people are interested in it.</li>
<li>we can report results in a few weeks</li>
</ul>
<p>Methods out there (in the conference call):</p>
<ul>
<li>CaVEMaN, DAP, CAVIAR</li>
<li>Can provide these results in a few weeks<ul>
<li>2MB radius</li>
<li>on Thyroid data</li>
<li>channel in Slack for this topic</li>
</ul>
</li>
</ul>
<p>General flavors</p>
<ul>
<li>Enrichment of eQTLs giving annotations: <ul>
<li>@william: Same model but various ways to estimate priors. Open to different prior options. Also see concensus results from different methods is also important.</li>
<li>But measuring consensus is difficult. Correlation is high but individual level there are problems.</li>
</ul>
</li>
<li>Conditional analysis: soon to be uploaded to <code>exchange</code></li>
<li>Additional type of integrated analysis<ul>
<li>@eric</li>
</ul>
</li>
</ul>
<p>Provide fine-mapping results for the community</p>
<h3 id="cell-specific-QTL">cell-specific QTL<a class="anchor-link" href="#cell-specific-QTL">&#182;</a></h3><ul>
<li>Fine-mapping for <strong>cell-specific</strong> QTL: cell-type deconvolution to get proportion of cells.<ul>
<li>interaction terms / interaction testing?</li>
</ul>
</li>
</ul>
<p>Provide fine-mapping results for the community</p>
<h3 id="Splicing-QTLs">Splicing QTLs<a class="anchor-link" href="#Splicing-QTLs">&#182;</a></h3><p>Can we fine-map splice QTLs?</p>
<h3 id="Interest-in-fine-mapping-companion-paper?">Interest in fine-mapping companion paper?<a class="anchor-link" href="#Interest-in-fine-mapping-companion-paper?">&#182;</a></h3><p>People are interested. There is also co-localization working group, but not progressing much.</p>
<h3 id="Interest-in-multi-tissue-fine-mapping?">Interest in multi-tissue fine-mapping?<a class="anchor-link" href="#Interest-in-multi-tissue-fine-mapping?">&#182;</a></h3><p>It is important but computationally hard. William is the only one who positively responed, but he only agree to do a few tissues.</p>
<h3 id="Other-related-stuff">Other related stuff<a class="anchor-link" href="#Other-related-stuff">&#182;</a></h3><ul>
<li>Sharing of results</li>
</ul>

</div>
</div>
</div>
<hr>
&copy 2015-2018 Gao Wang et al at Stephens Lab, University of Chicago
<p><small>Exported from <a href="http://github.com/stephenslab/gtex-eqtls/blob/529ff7dad746d8a51d4c9ba2ce663284535bf865/writeup/Meetings.ipynb"><code>writeup/Meetings.ipynb</code></a> committed by Gao Wang on Mon Jan 8 10:43:29 2018 <a href="http://github.com/stephenslab/gtex-eqtls/commit/529ff7dad746d8a51d4c9ba2ce663284535bf865">revision 5, 529ff7d</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
