<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.2" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="stylesheet" type="text/css" href="../css/jt.css">
<link rel="stylesheet" type="text/css" href="../css/toc2.css">
<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<link rel="stylesheet"
      href="../site_libs/highlight/textmate.css"
      type="text/css" />
<script src="../site_libs/highlight/highlight.js"></script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>
<script src="../js/toc2.js"></script>
<script src="../js/docs.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_HTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=writeupDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=writeupArray;
            var pos=writeupArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  var name=docs[a]
                  $('<li><a href="'+name+'.html"><font color="#073642"><b>'+name.replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  var name=docs[a]
                  $(".toc #toc-level0").append('<li><a href="'+name+'.html"><font color="#073642"><b>'+name.replace(/_/g," ")+'</b></font></a></li>');
            }
            $("#toc-header").hide();
    });
</script>
<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>
<title>GTEx eQTL</title>
<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>
<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">
<!-- code folding -->
<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">GTEx eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Overview</a>
</li>
<li>
  <a href="../analysis.html">Analysis</a>
</li>
<li>
  <a href="../prototype.html">Prototype</a>
</li>
<li>
  <a href="../writeup.html">Writeup</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/stephenslab/gtex-eqtls"> <span class="fa fa-github"></span> </a>
</li>
</ul>
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Meetings">Meetings<a class="anchor-link" href="#Meetings">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="20171211-GTEx-fine-mapping-workgroup-meeting">20171211 GTEx fine mapping workgroup meeting<a class="anchor-link" href="#20171211-GTEx-fine-mapping-workgroup-meeting">&#182;</a></h2><p>Phone conference with GTEx members from many cities ... Andrew presented.</p>
<p>Software to apply:</p>
<ul>
<li>Farhad Caviar: PIP on all tissues, 3 days</li>
<li>Andrew Caveman: PIP on all tissues, but whole blood first, a couple of days; all tissues 3 weeks</li>
<li>William DAP-G: a couple of days per tissue. A couple of tissues to start from.<ul>
<li>PIP</li>
<li>LD clusters</li>
</ul>
</li>
<li>Possibly 1 or 2 more software</li>
<li>Results for all tissues will be available mid-Jan.</li>
</ul>
<p>Data:</p>
<ul>
<li>Francois: VCF are almost final, though missingness are causing confusion (but in short we can mean-impute them). Should be finalized mid-week this week</li>
<li>Andrew: ChIP-seq benchmarks (an annotation list) to be shared. The data comes from roadmap epigenomics and will be lifted over to V9 data.</li>
<li>Tuuli: Has annotation for post transcriptional modification</li>
<li>William: Has annotation for TFBS.</li>
</ul>
<p>Simulation data:</p>
<ul>
<li>Andrew: not necessary. or at least should not be too close to the way any methods work. eg, no one-eqtl per gene type of setup.</li>
<li>William: is necessary. no need to worry about generating from certain models. Simulation based on simple linear model is generic enough.</li>
<li>Eleazar: Simulation results not interesting, compared to ChIP-seq / other annotation benchmarks.</li>
</ul>
<p>Exploration then benchmarking, or benchmarking right now?</p>
<ul>
<li>Blood results before Christmas, so that benchmarking can be put together.</li>
</ul>
<p>Benchmarking:</p>
<ul>
<li>Input: data and annotation. Annotation can be used for tests, or for validation<ul>
<li>Caveman now taks pre-computed annotations</li>
<li>Some type of Enrichment summary</li>
</ul>
</li>
<li>Output: scores?<ul>
<li>Gao has to figure out these</li>
<li>PIP, lfdr? at variant vs. gene level? how about LD cluster based methods?</li>
</ul>
</li>
<li>Blood results ready in a week to start putting together benchmarks. </li>
<li>All tissues by mid-Jan<ul>
<li>data availability</li>
</ul>
</li>
</ul>
<p>Extension of methods:</p>
<ul>
<li>cell-specific eQTLs</li>
<li>splicing eQTLs</li>
<li>age dependent QTLs</li>
<li>co-localisation<ul>
<li>Haky: GWAS group is also interested</li>
</ul>
</li>
<li>multiple traits<ul>
<li>Tuuli: can anyone put together a quick&amp;dirty M&amp;M analysis? <ul>
<li>Tuuli's group is developing something</li>
</ul>
</li>
<li>Andrew: pairwise tissue analysis</li>
<li>William: can look at a few tissues</li>
</ul>
</li>
</ul>
<h2 id="20171002-GTEx-fine-mapping-meeting">20171002 GTEx fine mapping meeting<a class="anchor-link" href="#20171002-GTEx-fine-mapping-meeting">&#182;</a></h2><p>Goal: methods do not repeat each-other; should build on each other.</p>
<h3 id="What-is-fine-mapping">What is fine-mapping<a class="anchor-link" href="#What-is-fine-mapping">&#182;</a></h3><ul>
<li>probability estimates of each variant being causal<ul>
<li>how to verify it? experimentally?</li>
</ul>
</li>
<li>Should use 1MB region?<ul>
<li>100KB should not lose a lot.</li>
</ul>
</li>
</ul>
<h3 id="Fine-mapping-methods">Fine-mapping methods<a class="anchor-link" href="#Fine-mapping-methods">&#182;</a></h3><p>Analyse/benchmark one data-set using all methods. The best tissue might be Thyroid to test it out.</p>
<ul>
<li>people are interested in it.</li>
<li>we can report results in a few weeks</li>
</ul>
<p>Methods out there (in the conference call):</p>
<ul>
<li>CaVEMaN, DAP, CAVIAR</li>
<li>Can provide these results in a few weeks<ul>
<li>2MB radius</li>
<li>on Thyroid data</li>
<li>channel in Slack for this topic</li>
</ul>
</li>
</ul>
<p>General flavors</p>
<ul>
<li>Enrichment of eQTLs giving annotations: <ul>
<li>@william: Same model but various ways to estimate priors. Open to different prior options. Also see concensus results from different methods is also important.</li>
<li>But measuring consensus is difficult. Correlation is high but individual level there are problems.</li>
</ul>
</li>
<li>Conditional analysis: soon to be uploaded to <code>exchange</code></li>
<li>Additional type of integrated analysis<ul>
<li>@eric</li>
</ul>
</li>
</ul>
<p>Provide fine-mapping results for the community</p>
<h3 id="cell-specific-QTL">cell-specific QTL<a class="anchor-link" href="#cell-specific-QTL">&#182;</a></h3><ul>
<li>Fine-mapping for <strong>cell-specific</strong> QTL: cell-type deconvolution to get proportion of cells.<ul>
<li>interaction terms / interaction testing?</li>
</ul>
</li>
</ul>
<p>Provide fine-mapping results for the community</p>
<h3 id="Splicing-QTLs">Splicing QTLs<a class="anchor-link" href="#Splicing-QTLs">&#182;</a></h3><p>Can we fine-map splice QTLs?</p>
<h3 id="Interest-in-fine-mapping-companion-paper?">Interest in fine-mapping companion paper?<a class="anchor-link" href="#Interest-in-fine-mapping-companion-paper?">&#182;</a></h3><p>People are interested. There is also co-localization working group, but not progressing much.</p>
<h3 id="Interest-in-multi-tissue-fine-mapping?">Interest in multi-tissue fine-mapping?<a class="anchor-link" href="#Interest-in-multi-tissue-fine-mapping?">&#182;</a></h3><p>It is important but computationally hard. William is the only one who positively responed, but he only agree to do a few tissues.</p>
<h3 id="Other-related-stuff">Other related stuff<a class="anchor-link" href="#Other-related-stuff">&#182;</a></h3><ul>
<li>Sharing of results</li>
</ul>
</div>
</div>
</div>
<hr>
&copy 2015-2017 Gao Wang et al at Stephens Lab, University of Chicago
<p><small>Exported from <a href="http://github.com/stephenslab/gtex-eqtls/blob/ca57c97387a8f9176cd718415398e8548316b84e/writeup/Meetings.ipynb"><code>writeup/Meetings.ipynb</code></a> committed by Gao Wang on Mon Dec 11 14:34:03 2017 <a href="http://github.com/stephenslab/gtex-eqtls/commit/ca57c97387a8f9176cd718415398e8548316b84e">revision 4, ca57c97</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
