<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=analysisDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=analysisArray;
            var docs_map=analysisArrayMap;
            var pos=analysisArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>GTEx V8 Multivariate Analysis</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">GTEx V8 Multivariate Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/gaow/mnm-gtex-v8"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Formatting-independent-eQTL-analysis-results-for-MASH-posterior-calculations">Formatting independent eQTL analysis results for MASH posterior calculations<a class="anchor-link" href="#Formatting-independent-eQTL-analysis-results-for-MASH-posterior-calculations">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>I have received from GTEx fine-mapping group independent eQTL data. Here I convert it to MASH input format in order to compute posterior.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Data-overview">Data overview<a class="anchor-link" href="#Data-overview">&#182;</a></h2><p>There are two sets of results, from conditional analysis and from DAP-G. Data format is very similar to <code>fastqtl</code> output, except that there is an additional first column of tissue name.</p>
<p>Here for each data-set take the union of cross tissue results and obtain a list of gene-SNP pair IDs in YAML format, load it to R, and extract directly from the HDF5 database the relevant information. 
I do it for batches of $K$ genes, obtaining roughly $\frac{30000}{K}$ batches.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[32]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="err">!</span> <span class="n">sos</span> <span class="n">run</span> <span class="n">Independent_eQTL_Results</span><span class="o">.</span><span class="n">ipynb</span> <span class="o">-</span><span class="n">h</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>usage: sos run Independent_eQTL_Results.ipynb
               [workflow_name | -t targets] [options] [workflow_options]
  workflow_name:        Single or combined workflows defined in this script
  targets:              One or more targets to generate
  options:              Single-hyphen sos parameters (see &#34;sos run -h&#34; for details)
  workflow_options:     Double-hyphen workflow-specific parameters

Workflows:
  default

Global Workflow Options:
  --data data/independent_eQTL/DAPG_pip_gt_0.01-AllTissues.txt (as path)
                        A text file containing data-set names
  --db fastqtl_to_mash_output/FastQTLSumStats.h5 (as path)
                        HDF5 file of the summary statistics database
  --batch-size 5000 (as int)
                        Batch size

Sections
  default_1:            Generate batches of gene-SNPs
  default_2:            Run data conversion
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Format-conversion-pipeline">Format conversion pipeline<a class="anchor-link" href="#Format-conversion-pipeline">&#182;</a></h2><p>Commands below processes the two fine-mapping results:</p>

<pre><code>sos run analysis/Independent_eQTL_Results.ipynb default:1 \
    --data data/independent_eQTL/DAPG_pip_gt_0.01-AllTissues.txt
sos run analysis/Independent_eQTL_Results.ipynb default:1 \
    --data data/independent_eQTL/ConditionalAnalysis_AllTissues.txt</code></pre>
<p>(and run on cluster:)</p>

<pre><code>sos run analysis/Independent_eQTL_Results.ipynb default:2 \
    --data data/independent_eQTL/DAPG_pip_gt_0.01-AllTissues.txt \
    -c data/fe961153.localhost.yml
sos run analysis/Independent_eQTL_Results.ipynb default:2 \
    --data data/independent_eQTL/ConditionalAnalysis_AllTissues.txt \
    -c data/fe961153.localhost.yml</code></pre>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="p">[</span><span class="k">global</span><span class="p">]</span>
<span class="c1"># A text file containing data-set names</span>
<span class="kn">parameter:</span> <span class="n">data</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="s2">&quot;data/independent_eQTL/DAPG_pip_gt_0.01-AllTissues.txt&quot;</span><span class="p">)</span>
<span class="c1"># HDF5 file of the summary statistics database</span>
<span class="kn">parameter:</span> <span class="n">db</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="s2">&quot;fastqtl_to_mash_output/FastQTLSumStats.h5&quot;</span><span class="p">)</span>
<span class="c1"># Batch size</span>
<span class="kn">parameter:</span> <span class="n">batch_size</span> <span class="o">=</span> <span class="mi">5000</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Workflow below extracts batches of YAML files of the format:</p>

<pre><code>gene_name_1: [snp1, snp2, ...]
gene_name_2: [snp1, snp2, ...]</code></pre>
<p>They will later be used to extract from the summary statistics database the gene-SNP pairs of interest.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Generate batches of gene-SNPs</span>
<span class="p">[</span><span class="n">default_1</span><span class="p">]</span>
<span class="n">fail_if</span><span class="p">(</span><span class="ow">not</span> <span class="n">data</span><span class="o">.</span><span class="n">is_file</span><span class="p">(),</span> <span class="n">msg</span> <span class="o">=</span> <span class="s1">&#39;Need data file!&#39;</span><span class="p">)</span>
<span class="kn">input:</span> <span class="n">data</span>
<span class="kn">output:</span> <span class="n">dynamic</span><span class="p">(</span><span class="n">f</span><span class="s2">&quot;{data:n}/{data:bn}.batch_*.yaml&quot;</span><span class="p">)</span>
<span class="kn">bash:</span> <span class="n">expand</span> <span class="o">=</span> <span class="bp">True</span>
    <span class="n">mkdir</span> <span class="o">-</span><span class="n">p</span> <span class="p">{</span><span class="n">data</span><span class="p">:</span><span class="n">n</span><span class="p">}</span>
<span class="kn">python:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s1">&#39;${ }&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{data:n}/{data:bn}.stdout&quot;</span>
    <span class="k">def</span> <span class="nf">write_res</span><span class="p">(</span><span class="n">res</span><span class="p">,</span> <span class="n">batch</span><span class="p">):</span>
        <span class="kn">import</span> <span class="nn">yaml</span>
        <span class="n">fn</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;${data:n}/${data:bn}.batch_{batch}.yaml&quot;</span>
        <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">fn</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">outfile</span><span class="p">:</span>
            <span class="n">yaml</span><span class="o">.</span><span class="n">dump</span><span class="p">(</span><span class="n">res</span><span class="p">,</span> <span class="n">outfile</span><span class="p">)</span>

    <span class="kn">import</span> <span class="nn">pandas</span> <span class="kn">as</span> <span class="nn">pd</span>
    <span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">:</span><span class="n">r</span><span class="p">},</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
    <span class="c1">#data.sort_values(by=[&#39;gene_id&#39;, &#39;variant_id&#39;], inplace=True)</span>
    <span class="n">res</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">()</span>
    <span class="n">idx</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="n">batch</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="k">print</span><span class="p">(</span><span class="s2">&quot;Start ...&quot;</span><span class="p">)</span>
    <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="nb">set</span><span class="p">(</span><span class="n">data</span><span class="p">[</span><span class="s1">&#39;gene_id&#39;</span><span class="p">]):</span>
        <span class="n">res</span><span class="p">[</span><span class="n">item</span><span class="o">.</span><span class="n">rsplit</span><span class="p">(</span><span class="s1">&#39;.&#39;</span><span class="p">,</span><span class="mi">1</span><span class="p">)[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">=</span> <span class="nb">sorted</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">data</span><span class="p">[</span><span class="s1">&#39;gene_id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">item</span><span class="p">][</span><span class="s1">&#39;variant_id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()))</span>
        <span class="k">if</span> <span class="n">idx</span> <span class="o">%</span> <span class="err">$</span><span class="p">{</span><span class="n">batch_size</span><span class="p">}</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="k">print</span><span class="p">(</span><span class="n">f</span><span class="s1">&#39;batch {batch}&#39;</span><span class="p">)</span>
            <span class="n">write_res</span><span class="p">(</span><span class="n">res</span><span class="p">,</span> <span class="n">batch</span><span class="p">)</span>
            <span class="n">batch</span> <span class="o">+=</span> <span class="mi">1</span>
            <span class="n">res</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">()</span>
        <span class="n">idx</span> <span class="o">+=</span> <span class="mi">1</span>
    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">res</span><span class="p">):</span>
        <span class="n">write_res</span><span class="p">(</span><span class="n">res</span><span class="p">,</span> <span class="n">batch</span><span class="p">)</span>
    <span class="k">print</span><span class="p">(</span><span class="s2">&quot;Done!&quot;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Workflow below extracts gene-SNP pairs of interest.</p>
<p><strong>FIXME: not all gene-SNPs in the YAML file exist in the summary statistics database!</strong>. Recall that the YAML file contains results from DAP-G / Conditional Regression analysis but I use their ID to extract from the original <code>fastqtl</code> results.</p>
<p>Possible reasons of inconsistency:</p>
<ol>
<li>The two data-sets analyzed slightly different gene-SNP combinations</li>
<li><strong>A bug in my <code>fastqtl</code> to summary database conversion code!</strong></li>
</ol>
<p>I saved these inconsistencies in <code>*.batch_*.stdout</code>. It prints which gene has fewer SNPs in the summary statistics DB. Most of them are just 1 SNP short. There are not many, anyways.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Run data conversion</span>
<span class="p">[</span><span class="n">default_2</span><span class="p">]</span>
<span class="kn">depends:</span> <span class="n">R_library</span><span class="p">(</span><span class="s1">&#39;rhdf5&#39;</span><span class="p">),</span> <span class="n">R_library</span><span class="p">(</span><span class="s1">&#39;yaml&#39;</span><span class="p">)</span>
<span class="kn">input:</span> <span class="n">f</span><span class="s2">&quot;{data:n}/{data:bn}.batch_*.yaml&quot;</span><span class="p">,</span> <span class="n">group_by</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">concurrent</span> <span class="o">=</span> <span class="bp">True</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s1">&#39;{_input:n}.rds&#39;</span>
<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">queue</span> <span class="o">=</span> <span class="s1">&#39;midway2&#39;</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;2h&#39;</span><span class="p">,</span> <span class="kp">trunk_size</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;4G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:bn}&#39;</span>
<span class="kn">R:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:n}.stdout&#39;</span>
    <span class="n">ConvertP2Z</span> <span class="o">&lt;-</span> <span class="n">function</span><span class="p">(</span><span class="n">pval</span><span class="p">,</span> <span class="n">beta</span><span class="p">)</span> <span class="p">{</span>
      <span class="n">z</span> <span class="o">&lt;-</span> <span class="nb">abs</span><span class="p">(</span><span class="n">qnorm</span><span class="p">(</span><span class="n">pval</span> <span class="o">/</span> <span class="mi">2</span><span class="p">))</span>
      <span class="n">z</span><span class="p">[</span><span class="n">which</span><span class="p">(</span><span class="n">beta</span> <span class="o">&lt;</span> <span class="mi">0</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="o">-</span><span class="mi">1</span> <span class="o">*</span> <span class="n">z</span><span class="p">[</span><span class="n">which</span><span class="p">(</span><span class="n">beta</span> <span class="o">&lt;</span> <span class="mi">0</span><span class="p">)]</span>
      <span class="k">return</span><span class="p">(</span><span class="n">z</span><span class="p">)</span>
    <span class="p">}</span>

    <span class="n">GetSS</span> <span class="o">&lt;-</span> <span class="n">function</span><span class="p">(</span><span class="n">table</span><span class="p">,</span> <span class="n">db</span><span class="p">)</span> <span class="p">{</span>
      <span class="n">dat</span> <span class="o">&lt;-</span> <span class="n">rhdf5</span><span class="p">::</span><span class="n">h5read</span><span class="p">(</span><span class="n">db</span><span class="p">,</span> <span class="n">table</span><span class="p">)</span>
      <span class="n">dat</span><span class="err">$</span><span class="s2">&quot;z&quot;</span> <span class="o">&lt;-</span> <span class="n">ConvertP2Z</span><span class="p">(</span><span class="n">dat</span><span class="err">$</span><span class="s2">&quot;pval&quot;</span><span class="p">,</span> <span class="n">dat</span><span class="err">$</span><span class="s2">&quot;beta&quot;</span><span class="p">)</span>
      <span class="k">for</span> <span class="p">(</span><span class="kp">name</span> <span class="ow">in</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;beta&quot;</span><span class="p">,</span> <span class="s2">&quot;se&quot;</span><span class="p">,</span> <span class="s2">&quot;pval&quot;</span><span class="p">,</span> <span class="s2">&quot;z&quot;</span><span class="p">))</span> <span class="p">{</span>
        <span class="n">dat</span><span class="p">[[</span><span class="kp">name</span><span class="p">]]</span> <span class="o">&lt;-</span> <span class="k">as</span><span class="o">.</span><span class="n">matrix</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">dat</span><span class="p">[[</span><span class="kp">name</span><span class="p">]]))</span>
        <span class="n">colnames</span><span class="p">(</span><span class="n">dat</span><span class="p">[[</span><span class="kp">name</span><span class="p">]])</span> <span class="o">&lt;-</span> <span class="n">dat</span><span class="err">$</span><span class="n">colnames</span>
        <span class="n">rownames</span><span class="p">(</span><span class="n">dat</span><span class="p">[[</span><span class="kp">name</span><span class="p">]])</span> <span class="o">&lt;-</span> <span class="n">dat</span><span class="err">$</span><span class="n">rownames</span>
      <span class="p">}</span>
      <span class="n">dat</span><span class="err">$</span><span class="n">colnames</span> <span class="o">&lt;-</span> <span class="n">dat</span><span class="err">$</span><span class="n">rownames</span> <span class="o">&lt;-</span> <span class="n">NULL</span>
      <span class="c1">## fastqtl tend to produce nan for betahat zero</span>
      <span class="c1">## here I set it to zero as well</span>
      <span class="n">dat</span><span class="p">[[</span><span class="s1">&#39;se&#39;</span><span class="p">]][</span><span class="n">which</span><span class="p">(</span><span class="ow">is</span><span class="o">.</span><span class="n">nan</span><span class="p">(</span><span class="n">dat</span><span class="p">[[</span><span class="s1">&#39;se&#39;</span><span class="p">]]))]</span> <span class="o">=</span> <span class="mi">0</span>
      <span class="k">return</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
    <span class="p">}</span>
    <span class="c1">##</span>
    <span class="n">meta</span> <span class="o">=</span> <span class="n">yaml</span><span class="p">::</span><span class="n">read_yaml</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
    <span class="n">res</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
    <span class="k">for</span> <span class="p">(</span><span class="n">gene</span> <span class="ow">in</span> <span class="n">names</span><span class="p">(</span><span class="n">meta</span><span class="p">))</span> <span class="p">{</span>
      <span class="n">snps</span> <span class="o">=</span> <span class="n">gsub</span><span class="p">(</span><span class="s1">&#39;^chr|_b38$&#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">,</span> <span class="n">meta</span><span class="p">[[</span><span class="n">gene</span><span class="p">]])</span>
      <span class="n">dat</span> <span class="o">=</span> <span class="n">GetSS</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">db</span><span class="p">:</span><span class="n">ar</span><span class="p">})</span>
      <span class="k">for</span> <span class="p">(</span><span class="n">p</span> <span class="ow">in</span> <span class="n">names</span><span class="p">(</span><span class="n">dat</span><span class="p">))</span> <span class="p">{</span>
        <span class="n">common_snps</span> <span class="o">=</span> <span class="n">intersect</span><span class="p">(</span><span class="n">snps</span><span class="p">,</span> <span class="n">rownames</span><span class="p">(</span><span class="n">dat</span><span class="p">[[</span><span class="n">p</span><span class="p">]]))</span>
        <span class="n">snp_diff</span> <span class="o">=</span> <span class="n">length</span><span class="p">(</span><span class="n">snps</span><span class="p">)</span> <span class="o">-</span> <span class="n">length</span><span class="p">(</span><span class="n">common_snps</span><span class="p">)</span>
        <span class="k">if</span> <span class="p">(</span><span class="n">snp_diff</span> <span class="o">&gt;</span> <span class="mi">0</span> <span class="o">&amp;&amp;</span> <span class="n">p</span> <span class="o">==</span> <span class="s1">&#39;beta&#39;</span><span class="p">)</span>
          <span class="k">print</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span> <span class="s1">&#39;is&#39;</span><span class="p">,</span> <span class="n">snp_diff</span><span class="p">,</span> <span class="s1">&#39;SNP short!&#39;</span><span class="p">))</span>
        <span class="k">if</span> <span class="p">(</span><span class="n">length</span><span class="p">(</span><span class="n">common_snps</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">)</span> <span class="p">{</span>
          <span class="k">if</span> <span class="p">(</span><span class="n">p</span> <span class="o">==</span> <span class="s1">&#39;beta&#39;</span><span class="p">)</span>
            <span class="k">print</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span> <span class="s1">&#39;is empty!&#39;</span><span class="p">))</span>
          <span class="nb">next</span>
        <span class="p">}</span>
        <span class="n">dat</span><span class="p">[[</span><span class="n">p</span><span class="p">]]</span> <span class="o">=</span> <span class="n">dat</span><span class="p">[[</span><span class="n">p</span><span class="p">]][</span><span class="n">common_snps</span><span class="p">,,</span><span class="n">drop</span><span class="o">=</span><span class="n">F</span><span class="p">]</span>
        <span class="n">rownames</span><span class="p">(</span><span class="n">dat</span><span class="p">[[</span><span class="n">p</span><span class="p">]])</span> <span class="o">=</span> <span class="n">paste0</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span> <span class="s1">&#39;_&#39;</span><span class="p">,</span> <span class="n">rownames</span><span class="p">(</span><span class="n">dat</span><span class="p">[[</span><span class="n">p</span><span class="p">]]))</span>
      <span class="p">}</span>
      <span class="k">if</span> <span class="p">(</span><span class="ow">is</span><span class="o">.</span><span class="n">null</span><span class="p">(</span><span class="n">res</span><span class="err">$</span><span class="n">Bhat</span><span class="p">))</span>
        <span class="n">res</span><span class="err">$</span><span class="n">Bhat</span> <span class="o">=</span> <span class="n">dat</span><span class="p">[[</span><span class="s1">&#39;beta&#39;</span><span class="p">]]</span>
      <span class="k">else</span>
        <span class="n">res</span><span class="err">$</span><span class="n">Bhat</span> <span class="o">=</span> <span class="n">rbind</span><span class="p">(</span><span class="n">res</span><span class="err">$</span><span class="n">Bhat</span><span class="p">,</span> <span class="n">dat</span><span class="p">[[</span><span class="s1">&#39;beta&#39;</span><span class="p">]])</span>
      <span class="k">if</span> <span class="p">(</span><span class="ow">is</span><span class="o">.</span><span class="n">null</span><span class="p">(</span><span class="n">res</span><span class="err">$</span><span class="n">Shat</span><span class="p">))</span>
        <span class="n">res</span><span class="err">$</span><span class="n">Shat</span> <span class="o">=</span> <span class="n">dat</span><span class="p">[[</span><span class="s1">&#39;se&#39;</span><span class="p">]]</span>
      <span class="k">else</span>
        <span class="n">res</span><span class="err">$</span><span class="n">Shat</span> <span class="o">=</span> <span class="n">rbind</span><span class="p">(</span><span class="n">res</span><span class="err">$</span><span class="n">Shat</span><span class="p">,</span> <span class="n">dat</span><span class="p">[[</span><span class="s1">&#39;se&#39;</span><span class="p">]])</span>
    <span class="p">}</span>
    <span class="n">saveRDS</span><span class="p">(</span><span class="n">res</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
</pre></div>

    </div>
</div>
</div>

</div>
<hr>
&copy 2018 Gao Wang, University of Chicago
<p><small>Exported from <a href="http://github.com/gaow/mnm-gtex-v8/blob/7b5baab45254850a4729e0f0204b32f68893a43a/analysis/Independent_eQTL_Results.ipynb"><code>analysis/Independent_eQTL_Results.ipynb</code></a> committed by Gao Wang on Wed Oct 30 14:41:42 2019 <a href="http://github.com/gaow/mnm-gtex-v8/commit/7b5baab45254850a4729e0f0204b32f68893a43a">revision 6, 7b5baab</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
