<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=analysisDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=analysisArray;
            var docs_map=analysisArrayMap;
            var pos=analysisArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>GTEx V8 Multivariate Analysis</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">GTEx V8 Multivariate Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/gaow/mnm-gtex-v8"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Mashing-the-GTEx-V8-release">Mashing the GTEx V8 release<a class="anchor-link" href="#Mashing-the-GTEx-V8-release">&#182;</a></h1><p>Here I run the latest <code>flashr + mashr</code> pipeline on the latest GTEx release.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p><em>Version: 2019.01.24 by Gao Wang and Yuxin Zou</em></p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[1]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="o">%</span><span class="n">revisions</span> <span class="o">-</span><span class="n">s</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>



<div class="output_html rendered_html output_subarea ">

        <table class="revision_table">
        <tr>
        <th>Revision</th>
        <th>Author</th>
        <th>Date</th>
        <th>Message</th>
        <tr>
        <tr><td><a target="_blank" href="git@github.com:stephenslab/gtexresults/blob/80e89a86f9ce380b7f30dc4dd64ef4ec632ef2d0/mashr_flashr_workflow.ipynb"><span class="revision_id">80e89a8<span></a></td>
<td>Gao Wang</td>
<td>2019-01-28</td>
<td>Add a note on HPC job submission</td></tr><tr><td><a target="_blank" href="git@github.com:stephenslab/gtexresults/blob/982a1b4cc4d8a14a77587fd55a825bf9ef00391e/mashr_flashr_workflow.ipynb"><span class="revision_id">982a1b4<span></a></td>
<td>Gao Wang</td>
<td>2019-01-28</td>
<td>Implement new $\hat{V}$ estimation method (with Yuxin Zou)</td></tr><tr><td><a target="_blank" href="git@github.com:stephenslab/gtexresults/blob/515e86951d09a0f2c1d4d3efde43882bfb75427e/mashr_flashr_workflow.ipynb"><span class="revision_id">515e869<span></a></td>
<td>Gao Wang</td>
<td>2018-11-22</td>
<td>Add a prompt for empty input posterior</td></tr><tr><td><a target="_blank" href="git@github.com:stephenslab/gtexresults/blob/ba9b20e5d705b20cda80826989293022452c3101/mashr_flashr_workflow.ipynb"><span class="revision_id">ba9b20e<span></a></td>
<td>Gao Wang</td>
<td>2018-11-22</td>
<td>Add posterior calculation for input 'strong' set</td></tr><tr><td><a target="_blank" href="git@github.com:stephenslab/gtexresults/blob/49494706a413999201362e72c9e20cbb95351be2/mashr_flashr_workflow.ipynb"><span class="revision_id">4949470<span></a></td>
<td>Gao Wang</td>
<td>2018-11-22</td>
<td>Fix mashr null correlation estimate interface</td></tr><tr><td><a target="_blank" href="git@github.com:stephenslab/gtexresults/blob/6145b3366b7850eac2e10bdd69634f6482e3359f/mashr_flashr_workflow.ipynb"><span class="revision_id">6145b33<span></a></td>
<td>Gao Wang</td>
<td>2018-11-21</td>
<td>Add --optmethod to configure convex optimization method to use</td></tr><tr><td><a target="_blank" href="git@github.com:stephenslab/gtexresults/blob/485381c8483bc4b770f8648b325a6386cd1a68d9/mashr_flashr_workflow.ipynb"><span class="revision_id">485381c<span></a></td>
<td>Gao Wang</td>
<td>2018-11-20</td>
<td>Fix mixSQP package name #2</td></tr><tr><td><a target="_blank" href="git@github.com:stephenslab/gtexresults/blob/c587791ce58a5d393782466ba762a32544edcf0e/mashr_flashr_workflow.ipynb"><span class="revision_id">c587791<span></a></td>
<td>Gao Wang</td>
<td>2018-11-20</td>
<td>Use the first cran release of mixSQP in default flashr workflow</td></tr><tr><td><a target="_blank" href="git@github.com:stephenslab/gtexresults/blob/092e20680cce52f594dd7fb95ee2c6b8ea85f036/mashr_flashr_workflow.ipynb"><span class="revision_id">092e206<span></a></td>
<td>Gao Wang</td>
<td>2018-09-24</td>
<td>Minor edits to documentation</td></tr><tr><td><a target="_blank" href="git@github.com:stephenslab/gtexresults/blob/4a5f2b78f94619f5759448e1fd0a9ba8fb600cb0/mashr_flashr_workflow.ipynb"><span class="revision_id">4a5f2b7<span></a></td>
<td>Gao Wang</td>
<td>2018-09-24</td>
<td>Add notes to results</td></tr><tr><td><a target="_blank" href="git@github.com:stephenslab/gtexresults/blob/e653d39708c6ba5b96da27472aed2896232814a8/mashr_flashr_workflow.ipynb"><span class="revision_id">e653d39<span></a></td>
<td>Gao Wang</td>
<td>2018-09-24</td>
<td>Configure posterior computation resources</td></tr><tr><td><a target="_blank" href="git@github.com:stephenslab/gtexresults/blob/ca56229760bd274d193689dbe9bf3efea8103b65/mashr_flashr_workflow.ipynb"><span class="revision_id">ca56229<span></a></td>
<td>Gao Wang</td>
<td>2018-09-23</td>
<td>Add posterior computation for input data-set</td></tr><tr><td><a target="_blank" href="git@github.com:stephenslab/gtexresults/blob/7db5500fd4549d39a21360fbcbf689bf5576094f/mashr_flashr_workflow.ipynb"><span class="revision_id">7db5500<span></a></td>
<td>Gao Wang</td>
<td>2018-09-23</td>
<td>Notes on GTEx V8 data conversion steps</td></tr><tr><td><a target="_blank" href="git@github.com:stephenslab/gtexresults/blob/68d2571e7d6284d02ac42d53e901968546298251/mashr_flashr_workflow.ipynb"><span class="revision_id">68d2571<span></a></td>
<td>Gao Wang</td>
<td>2018-05-30</td>
<td>Add mashr_flashr workflow</td></tr></table>
</div>

</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Data-overview">Data overview<a class="anchor-link" href="#Data-overview">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p><code>fastqtl</code> summary statistics data were obtained from dbGaP (data on CRI at UChicago Genetic Medicine). It has 49 tissues. [more description to come]</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Preparing-MASH-input">Preparing MASH input<a class="anchor-link" href="#Preparing-MASH-input">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Using an established workflow (which takes 33hrs to run on a cluster system as configured by <code>midway2.yml</code>; see inside <code>fastqtl_to_mash.ipynb</code> for a note on computing environment),</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">

<pre><code>INPUT_DIR=/project/compbio/GTEx_dbGaP/GTEx_Analysis_2017-06-05_v8/eqtl/GTEx_Analysis_v8_eQTL_all_associations
JOB_OPT="-c midway2.yml -q midway2"
sos run workflows/fastqtl_to_mash.ipynb --data-list $INPUT_DIR/FastQTLSumStats.list --common-suffix ".allpairs.txt" $JOB_OPT</code></pre>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>As a result of command above I obtained the "mashable" data-set in the same format <a href="https://stephenslab.github.io/gtexresults/gtexdata.html">as described here</a>.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Some-data-integrity-check">Some data integrity check<a class="anchor-link" href="#Some-data-integrity-check">&#182;</a></h3><ol>
<li>Check if I get the same number of groups (genes) at the end of HDF5 data conversion:</li>
</ol>

<pre><code>$ zcat Whole_Blood.allpairs.txt.gz | cut -f1 | sort -u | wc -l
20316
$ h5ls Whole_Blood.allpairs.txt.h5 | wc -l
20315</code></pre>
<p>The results agreed on Whole Blood sample (the original data has a header thus one line more than the H5 version). We should be good (since the pipeline reported success for all other files).</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Data-&amp;-job-summary">Data &amp; job summary<a class="anchor-link" href="#Data-&amp;-job-summary">&#182;</a></h3><p>The command above took 33 hours on UChicago RCC <code>midway2</code>.</p>

<pre><code>[MW] cat FastQTLSumStats.log
39832 out of 39832 groups merged!</code></pre>
<p>So we have a total of 39832 genes (union of 49 tissues).</p>

<pre><code>[MW] cat FastQTLSumStats.portable.log
15636 out of 39832 groups extracted!</code></pre>
<p>We have 15636 groups without missing data in any tissue. This will be used to train the MASH model.</p>
<p>The "mashable" data file is <code>FastQTLSumStats.mash.rds</code>, 124Mb serialized R file.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Multivariate-adaptive-shrinkage-(MASH)-analysis-of-eQTL-data">Multivariate adaptive shrinkage (MASH) analysis of eQTL data<a class="anchor-link" href="#Multivariate-adaptive-shrinkage-(MASH)-analysis-of-eQTL-data">&#182;</a></h2><p>Below is a "blackbox" implementation of the <code>mashr</code> eQTL workflow -- blackbox in the sense that you can run this pipeline as an executable, without thinking too much about it, if you see your problem fits our GTEx analysis scheme. However when reading it as a notebook it is a good source of information to help developing your own <code>mashr</code> analysis procedures.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Since the submission to biorxiv of Urbut 2017 we have improved implementation of MASH algorithm and made a new R package, <a href="https://github.com/stephenslab/mashr"><code>mashr</code></a>. Major improvements compared to Urbut 2017 are:</p>
<ol>
<li>Faster computation of likelihood and posterior quantities via matrix algebra tricks and a C++ implementation.</li>
<li>Faster computation of MASH mixture via convex optimization.</li>
<li>Replace <code>SFA</code> with <code>FLASH</code>, a new sparse factor analysis method to generate prior covariance candidates.</li>
<li>Improve estimate of residual variance $\hat{V}$.</li>
</ol>
<p>At this point, the input data have already been converted from the original eQTL summary statistics to a format convenient for analysis in MASH, as a result of running the data conversion pipeline in <code>fastqtl_to_mash.ipynb</code>.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Example command:</p>
<div class="highlight"><pre><span></span><span class="nv">JOB_OPT</span><span class="o">=</span><span class="s2">&quot;-j 8&quot;</span>
<span class="c1">#JOB_OPT=&quot;-c midway2.yml -q midway2&quot;</span>
sos run workflows/mashr_flashr_workflow.ipynb mash <span class="nv">$JOB_OPT</span> <span class="c1"># --data ... --cwd ... --vhat ...</span>
</pre></div>
<p><strong>FIXME: add comments on submitting jobs to HPC. Here we use the UChicago RCC cluster but other users can similarly configure their computating system to run the pipeline on HPC.</strong></p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Global-parameter-settings">Global parameter settings<a class="anchor-link" href="#Global-parameter-settings">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="p">[</span><span class="k">global</span><span class="p">]</span>
<span class="kn">parameter:</span> <span class="n">cwd</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="s1">&#39;./mashr_flashr_workflow_output&#39;</span><span class="p">)</span>
<span class="c1"># Input summary statistics data</span>
<span class="kn">parameter:</span> <span class="n">data</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="s2">&quot;fastqtl_to_mash_output/FastQTLSumStats.mash.rds&quot;</span><span class="p">)</span>
<span class="c1"># Prefix of output files. If not specified, it will derive it from data.</span>
<span class="c1"># If it is specified, for example, `--output-prefix AnalysisResults`</span>
<span class="c1"># It will save output files as `{cwd}/AnalysisResults*`.</span>
<span class="kn">parameter:</span> <span class="n">output_prefix</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>
<span class="c1"># Exchangable effect (EE) or exchangable z-scores (EZ)</span>
<span class="kn">parameter:</span> <span class="n">effect_model</span> <span class="o">=</span> <span class="s1">&#39;EZ&#39;</span>
<span class="c1"># Identifier of $\hat{V}$ estimate file</span>
<span class="c1"># It should be available as {cwd}/{data:bn}.V_{vhat}.rds</span>
<span class="kn">parameter:</span> <span class="n">vhat</span> <span class="o">=</span> <span class="s1">&#39;simple&#39;</span>
<span class="c1"># Additional label for vhat file</span>
<span class="kn">parameter:</span> <span class="n">vhat_file_label</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>
<span class="c1"># default method for convex optimization</span>
<span class="kn">parameter:</span> <span class="n">optmethod</span> <span class="o">=</span> <span class="s2">&quot;mixSQP&quot;</span>
<span class="c1"># Path to mosek license file, if `--optmethod mixIP` is used</span>
<span class="kn">parameter:</span> <span class="n">mosek_license</span> <span class="o">=</span> <span class="n">file_target</span><span class="p">(</span><span class="s2">&quot;~/.mosek.lic&quot;</span><span class="p">)</span>
<span class="c1"># Most established heavy-lifting computations are in C++ via Rcpp</span>
<span class="c1"># but some experimental features are in R. This argument allows one</span>
<span class="c1"># to switch between implementations</span>
<span class="kn">parameter:</span> <span class="n">implementation</span> <span class="o">=</span> <span class="s1">&#39;Rcpp&#39;</span>
<span class="c1"># Number of components in PCA analysis for prior</span>
<span class="c1"># default to 3 as in mash paper</span>
<span class="c1"># set it to 0 to not use PCA priors</span>
<span class="kn">parameter:</span> <span class="n">npc</span> <span class="o">=</span> <span class="mi">3</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">absolute</span><span class="p">()</span>
<span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">output_prefix</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
    <span class="n">output_prefix</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{data:bn}&quot;</span>
<span class="k">if</span> <span class="n">vhat_file_label</span> <span class="ow">and</span> <span class="ow">not</span> <span class="n">vhat_file_label</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s1">&#39;_&#39;</span><span class="p">):</span>
    <span class="n">vhat_file_label</span> <span class="o">=</span> <span class="s1">&#39;_&#39;</span> <span class="o">+</span> <span class="n">vhat_file_label</span>
<span class="n">flash_data</span> <span class="o">=</span> <span class="n">file_target</span><span class="p">(</span><span class="n">f</span><span class="s2">&quot;{cwd:a}/{output_prefix}.{effect_model}.flash.rds&quot;</span><span class="p">)</span>
<span class="n">prior_data</span> <span class="o">=</span> <span class="n">file_target</span><span class="p">(</span><span class="n">f</span><span class="s2">&quot;{cwd:a}/{output_prefix}.{effect_model}.FL_PC{npc}.rds&quot;</span><span class="p">)</span>
<span class="n">vhat_data</span> <span class="o">=</span> <span class="n">file_target</span><span class="p">(</span><span class="n">f</span><span class="s2">&quot;{cwd:a}/{output_prefix}.{effect_model}.FL_PC{npc}.V_{vhat}{vhat_file_label}.rds&quot;</span><span class="p">)</span>
<span class="n">fail_if</span><span class="p">(</span><span class="n">optmethod</span> <span class="o">==</span> <span class="s2">&quot;mixIP&quot;</span> <span class="ow">and</span> <span class="ow">not</span> <span class="n">mosek_license</span><span class="o">.</span><span class="n">is_file</span><span class="p">(),</span> <span class="n">msg</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;To use mixIP optimization, please put a valid copy (NOT a symbolic link!) of MOSEK license to: </span><span class="se">\n</span><span class="s1">``{mosek_license}``&#39;</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">sort_uniq</span><span class="p">(</span><span class="n">seq</span><span class="p">):</span>
    <span class="n">seen</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>
    <span class="k">return</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">seq</span> <span class="k">if</span> <span class="ow">not</span> <span class="p">(</span><span class="n">x</span> <span class="ow">in</span> <span class="n">seen</span> <span class="ow">or</span> <span class="n">seen</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">x</span><span class="p">))]</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Command-interface">Command interface<a class="anchor-link" href="#Command-interface">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">sos</span> <span class="n">run</span> <span class="n">mashr_flashr_workflow</span><span class="o">.</span><span class="n">ipynb</span> <span class="o">-</span><span class="n">h</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>usage: sos run mashr_flashr_workflow.ipynb
               [workflow_name | -t targets] [options] [workflow_options]
  workflow_name:        Single or combined workflows defined in this script
  targets:              One or more targets to generate
  options:              Single-hyphen sos parameters (see &#34;sos run -h&#34; for details)
  workflow_options:     Double-hyphen workflow-specific parameters

Workflows:
  flash
  prior
  vhat_identity
  vhat_simple
  vhat_mle
  vhat_corshrink_xcondition
  vhat_simple_specific
  mash
  posterior

Global Workflow Options:
  --cwd mashr_flashr_workflow_output (as path)
  --data fastqtl_to_mash_output/FastQTLSumStats.mash.rds (as path)
                        Input summary statistics data
  --output-prefix &#39;&#39;
                        Prefix of output files. If not specified, it will derive
                        it from data. If it is specified, for example,
                        `--output-prefix AnalysisResults` It will save output
                        files as `{cwd}/AnalysisResults*`.
  --effect-model EZ
                        Exchangable effect (EE) or exchangable z-scores (EZ)
  --vhat simple
                        Identifier of $\hat{V}$ estimate file It should be
                        available as {cwd}/{data:bn}.V_{vhat}.rds
  --vhat-file-label &#39;&#39;
                        Additional label for vhat file
  --optmethod mixSQP
                        default method for convex optimization
  --mosek-license /home/gaow/.mosek.lic (as file_target)
                        Path to mosek license file, if `--optmethod mixIP` is
                        used
  --implementation Rcpp
                        Most established heavy-lifting computations are in C++
                        via Rcpp but some experimental features are in R. This
                        argument allows one to switch between implementations
  --npc 3 (as int)
                        Number of components in PCA analysis for prior default
                        to 3 as in mash paper set it to 0 to not use PCA priors

Sections
  flash:                Perform FLASH analysis (time estimate: 20min)
  prior:                Compute data-driven / canonical prior matrices (time
                        estimate: 2h ~ 12h for ~30 49 by 49 matrix mixture)
  vhat_identity:        V estimate: &#34;identity&#34; method
  vhat_simple:          V estimate: &#34;simple&#34; method (using null z-scores)
  vhat_mle:             V estimate: &#34;mle&#34; method
    Workflow Options:
      --n-subset 6000 (as int)
                        number of samples to use
      --max-iter 6 (as int)
                        maximum number of iterations
  vhat_corshrink_xcondition_1: Estimate each V separately via corshrink
    Workflow Options:
      --util-script /project/mstephens/gtex/scripts/SumstatQuery.R (as path)
                        Utility script
      --gene-list . (as path)
                        List of genes to analyze
  vhat_corshrink_xcondition_2, vhat_simple_specific_2: Consolidate Vhat into one
                        file
    Workflow Options:
      --gene-list . (as path)
                        List of genes to analyze
  vhat_simple_specific_1: Estimate each V separately via &#34;simple&#34; method
    Workflow Options:
      --util-script /project/mstephens/gtex/scripts/SumstatQuery.R (as path)
                        Utility script
      --gene-list . (as path)
                        List of genes to analyze
  mash_1:               Fit MASH mixture model (time estimate: &lt;15min for 70K by
                        49 matrix)
  mash_2:               Compute posterior for the &#34;strong&#34; set of data as in
                        Urbut et al 2017. This is optional because most of the
                        time we want to apply the MASH model learned on much
                        larger data-set.
    Workflow Options:
      --[no-]compute-posterior (default to True)
                        default to True; use --no-compute-posterior to disable
                        this
      --posterior-vhat-file . (as path)
                        input Vhat file for the batch of posterior data
  posterior:            Apply posterior calculations
    Workflow Options:
      --mash-model  path(f&#34;{vhat_data:n}.mash_model.rds&#34;)

      --posterior-input  paths()

      --posterior-vhat-files  paths()

      --data-table-name &#39;&#39;
                        eg, if data is saved in R list as data$strong, then when
                        you specify `--data-table-name strong` it will read the
                        data as readRDS(&#39;{_input:r}&#39;)$strong
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Compute-MASH-priors">Compute MASH priors<a class="anchor-link" href="#Compute-MASH-priors">&#182;</a></h3><p>Main reference are our <code>mashr</code> vignettes <a href="https://stephenslab.github.io/mashr/articles/eQTL_outline.html">this for mashr eQTL outline</a> and <a href="https://github.com/stephenslab/mashr/blob/master/vignettes/flash_mash.Rmd">this for using FLASH prior</a>. 
The latter was written recently specifically for this effort, and will likely be subject to changes for future versions.</p>
<p>If you use <code>--optmethod mixIP</code> you will have to put a copy of <a href="https://www.mosek.com/products/academic-licenses">MOSEK license file</a> to <code>&lt;workdir&gt;/mosek.lic</code> (ie, <code>mashr_flashr_workflow_output/mosek.lic</code> if you did not change any settings below). Current default method is <code>mixSQP</code>.</p>
<p>The outcome of this workflow should be found under <code>./mashr_flashr_workflow_output</code> folder (can be configured). File names have pattern <code>*.mash_model_*.rds</code>. They can be used to computer posterior for input list of gene-SNP pairs (see next section).</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="flashr-prior-covariances"><code>flashr</code> prior covariances<a class="anchor-link" href="#flashr-prior-covariances">&#182;</a></h4>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[4]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Perform FLASH analysis (time estimate: 20min)</span>
<span class="p">[</span><span class="n">flash</span><span class="p">]</span>
<span class="kn">depends:</span> <span class="n">R_library</span><span class="p">(</span><span class="s2">&quot;flashr&quot;</span><span class="p">)</span>
<span class="kn">input:</span> <span class="n">data</span>
<span class="kn">output:</span> <span class="n">flash_data</span>

<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;2h&#39;</span><span class="p">,</span> <span class="kp">trunk_size</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;4G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:bn}&#39;</span>
<span class="kn">R:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="kp">workdir</span> <span class="o">=</span> <span class="n">cwd</span><span class="p">,</span> <span class="kp">env</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;MOSEKLM_LICENSE_FILE&#39;</span><span class="p">:</span> <span class="nb">str</span><span class="p">(</span><span class="n">mosek_license</span><span class="p">)},</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stderr&quot;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stdout&quot;</span>
    <span class="n">library</span><span class="p">(</span><span class="n">flashr</span><span class="p">)</span>
    <span class="n">library</span><span class="p">(</span><span class="n">mixsqp</span><span class="p">)</span>
    <span class="n">library</span><span class="p">(</span><span class="n">mashr</span><span class="p">)</span>
    
    <span class="n">my_init_fn</span> <span class="o">&lt;-</span> <span class="n">function</span><span class="p">(</span><span class="n">Y</span><span class="p">,</span> <span class="n">K</span> <span class="o">=</span> <span class="mi">1</span><span class="p">)</span> <span class="p">{</span>
      <span class="n">ret</span> <span class="o">=</span> <span class="n">flashr</span><span class="p">:::</span><span class="n">udv_si</span><span class="p">(</span><span class="n">Y</span><span class="p">,</span> <span class="n">K</span><span class="p">)</span>
      <span class="n">pos_sum</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">ret</span><span class="err">$</span><span class="n">v</span><span class="p">[</span><span class="n">ret</span><span class="err">$</span><span class="n">v</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">])</span>
      <span class="n">neg_sum</span> <span class="o">=</span> <span class="o">-</span><span class="nb">sum</span><span class="p">(</span><span class="n">ret</span><span class="err">$</span><span class="n">v</span><span class="p">[</span><span class="n">ret</span><span class="err">$</span><span class="n">v</span> <span class="o">&lt;</span> <span class="mi">0</span><span class="p">])</span>
      <span class="k">if</span> <span class="p">(</span><span class="n">neg_sum</span> <span class="o">&gt;</span> <span class="n">pos_sum</span><span class="p">)</span> <span class="p">{</span>
        <span class="k">return</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">u</span> <span class="o">=</span> <span class="o">-</span><span class="n">ret</span><span class="err">$</span><span class="n">u</span><span class="p">,</span> <span class="n">d</span> <span class="o">=</span> <span class="n">ret</span><span class="err">$</span><span class="n">d</span><span class="p">,</span> <span class="n">v</span> <span class="o">=</span> <span class="o">-</span><span class="n">ret</span><span class="err">$</span><span class="n">v</span><span class="p">))</span>
      <span class="p">}</span> <span class="k">else</span>
      <span class="k">return</span><span class="p">(</span><span class="n">ret</span><span class="p">)</span>
    <span class="p">}</span>

    <span class="n">flash_pipeline</span> <span class="o">=</span> <span class="n">function</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="o">...</span><span class="p">)</span> <span class="p">{</span>
      <span class="c1">## current state-of-the art</span>
      <span class="c1">## suggested by Jason Willwerscheid</span>
      <span class="c1">## cf: discussion section of</span>
      <span class="c1">## https://willwerscheid.github.io/MASHvFLASH/MASHvFLASHnn2.html</span>
      <span class="n">ebnm_fn</span> <span class="o">=</span> <span class="s2">&quot;ebnm_ash&quot;</span>
      <span class="n">ebnm_param</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">l</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">mixcompdist</span> <span class="o">=</span> <span class="s2">&quot;normal&quot;</span><span class="p">,</span>
                               <span class="n">optmethod</span> <span class="o">=</span> <span class="s2">&quot;${optmethod}&quot;</span><span class="p">),</span>
                        <span class="n">f</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">mixcompdist</span> <span class="o">=</span> <span class="s2">&quot;+uniform&quot;</span><span class="p">,</span>
                               <span class="n">optmethod</span> <span class="o">=</span> <span class="s2">&quot;${optmethod}&quot;</span><span class="p">))</span>
      <span class="c1">##</span>
      <span class="n">fl_g</span> <span class="o">&lt;-</span> <span class="n">flashr</span><span class="p">:::</span><span class="n">flash_greedy_workhorse</span><span class="p">(</span><span class="n">data</span><span class="p">,</span>
                    <span class="n">var_type</span> <span class="o">=</span> <span class="s2">&quot;constant&quot;</span><span class="p">,</span>
                    <span class="n">ebnm_fn</span> <span class="o">=</span> <span class="n">ebnm_fn</span><span class="p">,</span>
                    <span class="n">ebnm_param</span> <span class="o">=</span> <span class="n">ebnm_param</span><span class="p">,</span>
                    <span class="n">init_fn</span> <span class="o">=</span> <span class="s2">&quot;my_init_fn&quot;</span><span class="p">,</span>
                    <span class="n">stopping_rule</span> <span class="o">=</span> <span class="s2">&quot;factors&quot;</span><span class="p">,</span>
                    <span class="n">tol</span> <span class="o">=</span> <span class="mf">1e-3</span><span class="p">,</span>
                    <span class="n">verbose_output</span> <span class="o">=</span> <span class="s2">&quot;odF&quot;</span><span class="p">)</span>
      <span class="n">fl_b</span> <span class="o">&lt;-</span> <span class="n">flashr</span><span class="p">:::</span><span class="n">flash_backfit_workhorse</span><span class="p">(</span><span class="n">data</span><span class="p">,</span>
                    <span class="n">f_init</span> <span class="o">=</span> <span class="n">fl_g</span><span class="p">,</span>
                    <span class="n">var_type</span> <span class="o">=</span> <span class="s2">&quot;constant&quot;</span><span class="p">,</span>
                    <span class="n">ebnm_fn</span> <span class="o">=</span> <span class="n">ebnm_fn</span><span class="p">,</span>
                    <span class="n">ebnm_param</span> <span class="o">=</span> <span class="n">ebnm_param</span><span class="p">,</span>
                    <span class="n">stopping_rule</span> <span class="o">=</span> <span class="s2">&quot;factors&quot;</span><span class="p">,</span>
                    <span class="n">tol</span> <span class="o">=</span> <span class="mf">1e-3</span><span class="p">,</span>
                    <span class="n">verbose_output</span> <span class="o">=</span> <span class="s2">&quot;odF&quot;</span><span class="p">)</span>
      <span class="k">return</span><span class="p">(</span><span class="n">fl_b</span><span class="p">)</span>
    <span class="p">}</span>

    <span class="n">cov_flash</span> <span class="o">=</span> <span class="n">function</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">subset</span> <span class="o">=</span> <span class="n">NULL</span><span class="p">,</span> <span class="n">non_canonical</span> <span class="o">=</span> <span class="n">FALSE</span><span class="p">,</span> <span class="n">save_model</span> <span class="o">=</span> <span class="n">NULL</span><span class="p">)</span> <span class="p">{</span>
      <span class="k">if</span><span class="p">(</span><span class="ow">is</span><span class="o">.</span><span class="n">null</span><span class="p">(</span><span class="n">subset</span><span class="p">))</span> <span class="n">subset</span> <span class="o">=</span> <span class="mi">1</span><span class="p">:</span><span class="n">mashr</span><span class="p">:::</span><span class="n">n_effects</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
      <span class="n">b</span><span class="o">.</span><span class="n">center</span> <span class="o">=</span> <span class="nb">apply</span><span class="p">(</span><span class="n">data</span><span class="err">$</span><span class="n">Bhat</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="n">x</span> <span class="o">-</span> <span class="n">mean</span><span class="p">(</span><span class="n">x</span><span class="p">))</span>
      <span class="c1">## Only keep factors with at least two values greater than 1 / sqrt(n)</span>
      <span class="n">find_nonunique_effects</span> <span class="o">&lt;-</span> <span class="n">function</span><span class="p">(</span><span class="n">fl</span><span class="p">)</span> <span class="p">{</span>
        <span class="n">thresh</span> <span class="o">&lt;-</span> <span class="mi">1</span><span class="o">/</span><span class="n">sqrt</span><span class="p">(</span><span class="n">ncol</span><span class="p">(</span><span class="n">fl</span><span class="err">$</span><span class="n">fitted_values</span><span class="p">))</span>
        <span class="n">vals_above_avg</span> <span class="o">&lt;-</span> <span class="n">colSums</span><span class="p">(</span><span class="n">fl</span><span class="err">$</span><span class="n">ldf</span><span class="err">$</span><span class="n">f</span> <span class="o">&gt;</span> <span class="n">thresh</span><span class="p">)</span>
        <span class="n">nonuniq_effects</span> <span class="o">&lt;-</span> <span class="n">which</span><span class="p">(</span><span class="n">vals_above_avg</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">)</span>
        <span class="k">return</span><span class="p">(</span><span class="n">fl</span><span class="err">$</span><span class="n">ldf</span><span class="err">$</span><span class="n">f</span><span class="p">[,</span> <span class="n">nonuniq_effects</span><span class="p">,</span> <span class="n">drop</span> <span class="o">=</span> <span class="n">FALSE</span><span class="p">])</span>
      <span class="p">}</span>

      <span class="n">fmodel</span> <span class="o">=</span> <span class="n">flash_pipeline</span><span class="p">(</span><span class="n">b</span><span class="o">.</span><span class="n">center</span><span class="p">)</span>
      <span class="k">if</span> <span class="p">(</span><span class="n">non_canonical</span><span class="p">)</span>
          <span class="n">flash_f</span> <span class="o">=</span> <span class="n">find_nonunique_effects</span><span class="p">(</span><span class="n">fmodel</span><span class="p">)</span>
      <span class="k">else</span> 
          <span class="n">flash_f</span> <span class="o">=</span> <span class="n">fmodel</span><span class="err">$</span><span class="n">ldf</span><span class="err">$</span><span class="n">f</span>
      <span class="c1">## row.names(flash_f) = colnames(b)</span>
      <span class="k">if</span> <span class="p">(</span><span class="err">!</span><span class="ow">is</span><span class="o">.</span><span class="n">null</span><span class="p">(</span><span class="n">save_model</span><span class="p">))</span> <span class="n">saveRDS</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">model</span><span class="o">=</span><span class="n">fmodel</span><span class="p">,</span> <span class="n">factors</span><span class="o">=</span><span class="n">flash_f</span><span class="p">),</span> <span class="n">save_model</span><span class="p">)</span>
      <span class="k">if</span><span class="p">(</span><span class="n">ncol</span><span class="p">(</span><span class="n">flash_f</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">){</span>
        <span class="n">U</span><span class="o">.</span><span class="n">flash</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="s2">&quot;tFLASH&quot;</span> <span class="o">=</span> <span class="n">t</span><span class="p">(</span><span class="n">fmodel</span><span class="err">$</span><span class="n">fitted_values</span><span class="p">)</span> <span class="o">%*%</span> <span class="n">fmodel</span><span class="err">$</span><span class="n">fitted_values</span> <span class="o">/</span> <span class="n">nrow</span><span class="p">(</span><span class="n">fmodel</span><span class="err">$</span><span class="n">fitted_values</span><span class="p">))</span>
      <span class="p">}</span> <span class="k">else</span><span class="p">{</span>
        <span class="n">U</span><span class="o">.</span><span class="n">flash</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="n">cov_from_factors</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="k">as</span><span class="o">.</span><span class="n">matrix</span><span class="p">(</span><span class="n">flash_f</span><span class="p">)),</span> <span class="s2">&quot;FLASH&quot;</span><span class="p">),</span>
                    <span class="nb">list</span><span class="p">(</span><span class="s2">&quot;tFLASH&quot;</span> <span class="o">=</span> <span class="n">t</span><span class="p">(</span><span class="n">fmodel</span><span class="err">$</span><span class="n">fitted_values</span><span class="p">)</span> <span class="o">%*%</span> <span class="n">fmodel</span><span class="err">$</span><span class="n">fitted_values</span> <span class="o">/</span> <span class="n">nrow</span><span class="p">(</span><span class="n">fmodel</span><span class="err">$</span><span class="n">fitted_values</span><span class="p">)))</span>
      <span class="p">}</span>
      <span class="k">return</span><span class="p">(</span><span class="n">U</span><span class="o">.</span><span class="n">flash</span><span class="p">)</span>
    <span class="p">}</span>
    <span class="c1">##</span>
    <span class="n">dat</span> <span class="o">=</span> <span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
    <span class="n">dat</span> <span class="o">=</span> <span class="n">mash_set_data</span><span class="p">(</span><span class="n">dat</span><span class="err">$</span><span class="n">strong</span><span class="o">.</span><span class="n">b</span><span class="p">,</span> <span class="n">dat</span><span class="err">$</span><span class="n">strong</span><span class="o">.</span><span class="n">s</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="mi">1</span> <span class="k">if</span> <span class="n">effect_model</span> <span class="o">==</span> <span class="s1">&#39;EZ&#39;</span> <span class="k">else</span> <span class="mi">0</span><span class="p">})</span>
    <span class="n">res</span> <span class="o">=</span> <span class="n">cov_flash</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="n">non_canonical</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">save_model</span> <span class="o">=</span> <span class="s2">&quot;${_output:n}.model.rds&quot;</span><span class="p">)</span>
    <span class="n">saveRDS</span><span class="p">(</span><span class="n">res</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Other-priors-and-refinement-via-Extreme-Deconvolution">Other priors and refinement via Extreme Deconvolution<a class="anchor-link" href="#Other-priors-and-refinement-via-Extreme-Deconvolution">&#182;</a></h4>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[5]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Compute data-driven / canonical prior matrices (time estimate: 2h ~ 12h for ~30 49 by 49 matrix mixture)</span>
<span class="p">[</span><span class="n">prior</span><span class="p">]</span>
<span class="kn">depends:</span> <span class="n">R_library</span><span class="p">(</span><span class="s2">&quot;mashr&quot;</span><span class="p">)</span>
<span class="kn">input:</span> <span class="n">data</span><span class="p">,</span> <span class="n">output_from</span><span class="p">(</span><span class="s1">&#39;flash&#39;</span><span class="p">)</span>
<span class="kn">output:</span> <span class="n">prior_data</span>

<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;36h&#39;</span><span class="p">,</span> <span class="kp">trunk_size</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;4G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:bn}&#39;</span>
<span class="kn">R:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="kp">workdir</span> <span class="o">=</span> <span class="n">cwd</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stderr&quot;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stdout&quot;</span>
    <span class="n">library</span><span class="p">(</span><span class="n">mashr</span><span class="p">)</span>
    <span class="n">dat</span> <span class="o">=</span> <span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">})</span>
    <span class="n">mash_data</span> <span class="o">=</span> <span class="n">mash_set_data</span><span class="p">(</span><span class="n">dat</span><span class="err">$</span><span class="n">strong</span><span class="o">.</span><span class="n">b</span><span class="p">,</span> <span class="n">Shat</span><span class="o">=</span><span class="n">dat</span><span class="err">$</span><span class="n">strong</span><span class="o">.</span><span class="n">s</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="mi">1</span> <span class="k">if</span> <span class="n">effect_model</span> <span class="o">==</span> <span class="s1">&#39;EZ&#39;</span> <span class="k">else</span> <span class="mi">0</span><span class="p">})</span>
    <span class="c1"># FLASH matrices</span>
    <span class="n">U</span><span class="o">.</span><span class="n">flash</span> <span class="o">=</span> <span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">1</span><span class="p">]:</span><span class="n">r</span><span class="p">})</span>
    <span class="c1"># SVD matrices</span>
    <span class="n">U</span><span class="o">.</span><span class="n">pca</span> <span class="o">=</span> <span class="err">$</span><span class="p">{</span><span class="s2">&quot;cov_pca(mash_data, </span><span class="si">%s</span><span class="s2">)&quot;</span> <span class="o">%</span> <span class="n">npc</span> <span class="k">if</span> <span class="n">npc</span> <span class="o">&gt;</span> <span class="mi">0</span> <span class="k">else</span> <span class="s2">&quot;list()&quot;</span><span class="p">}</span>
    <span class="c1"># Emperical cov matrix</span>
    <span class="n">X</span><span class="o">.</span><span class="n">center</span> <span class="o">=</span> <span class="nb">apply</span><span class="p">(</span><span class="n">mash_data</span><span class="err">$</span><span class="n">Bhat</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="n">x</span> <span class="o">-</span> <span class="n">mean</span><span class="p">(</span><span class="n">x</span><span class="p">))</span>
    <span class="c1"># Denoised data-driven matrices</span>
    <span class="n">U</span><span class="o">.</span><span class="n">ed</span> <span class="o">=</span> <span class="n">cov_ed</span><span class="p">(</span><span class="n">mash_data</span><span class="p">,</span> <span class="n">c</span><span class="p">(</span><span class="n">U</span><span class="o">.</span><span class="n">flash</span><span class="p">,</span> <span class="n">U</span><span class="o">.</span><span class="n">pca</span><span class="p">,</span> <span class="nb">list</span><span class="p">(</span><span class="s2">&quot;XX&quot;</span> <span class="o">=</span> <span class="n">t</span><span class="p">(</span><span class="n">X</span><span class="o">.</span><span class="n">center</span><span class="p">)</span> <span class="o">%*%</span> <span class="n">X</span><span class="o">.</span><span class="n">center</span> <span class="o">/</span> <span class="n">nrow</span><span class="p">(</span><span class="n">X</span><span class="o">.</span><span class="n">center</span><span class="p">))),</span> <span class="kp">logfile</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">nr</span><span class="p">})</span>
    <span class="c1"># Canonical matrices</span>
    <span class="n">U</span><span class="o">.</span><span class="n">can</span> <span class="o">=</span> <span class="n">cov_canonical</span><span class="p">(</span><span class="n">mash_data</span><span class="p">)</span>
    <span class="n">saveRDS</span><span class="p">(</span><span class="n">c</span><span class="p">(</span><span class="n">U</span><span class="o">.</span><span class="n">ed</span><span class="p">,</span> <span class="n">U</span><span class="o">.</span><span class="n">can</span><span class="p">),</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Estimate-residual-variance">Estimate residual variance<a class="anchor-link" href="#Estimate-residual-variance">&#182;</a></h3><p>FIXME: add some narratives here explaining what we do in each method.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[6]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># V estimate: &quot;identity&quot; method</span>
<span class="p">[</span><span class="n">vhat_identity</span><span class="p">]</span>
<span class="kn">input:</span> <span class="n">data</span>
<span class="kn">output:</span> <span class="n">vhat_data</span>
<span class="kn">R:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="kp">workdir</span> <span class="o">=</span> <span class="n">cwd</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stderr&quot;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stdout&quot;</span>
    <span class="n">dat</span> <span class="o">=</span> <span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
    <span class="n">saveRDS</span><span class="p">(</span><span class="n">diag</span><span class="p">(</span><span class="n">ncol</span><span class="p">(</span><span class="n">dat</span><span class="err">$</span><span class="n">random</span><span class="o">.</span><span class="n">b</span><span class="p">)),</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[7]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># V estimate: &quot;simple&quot; method (using null z-scores)</span>
<span class="p">[</span><span class="n">vhat_simple</span><span class="p">]</span>
<span class="kn">depends:</span> <span class="n">R_library</span><span class="p">(</span><span class="s2">&quot;mashr&quot;</span><span class="p">)</span>
<span class="kn">input:</span> <span class="n">data</span>
<span class="kn">output:</span> <span class="n">vhat_data</span>
<span class="kn">R:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="kp">workdir</span> <span class="o">=</span> <span class="n">cwd</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stderr&quot;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stdout&quot;</span>
    <span class="n">library</span><span class="p">(</span><span class="n">mashr</span><span class="p">)</span>
    <span class="n">dat</span> <span class="o">=</span> <span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
    <span class="n">vhat</span> <span class="o">=</span> <span class="n">estimate_null_correlation_simple</span><span class="p">(</span><span class="n">mash_set_data</span><span class="p">(</span><span class="n">dat</span><span class="err">$</span><span class="n">random</span><span class="o">.</span><span class="n">b</span><span class="p">,</span> <span class="n">Shat</span><span class="o">=</span><span class="n">dat</span><span class="err">$</span><span class="n">random</span><span class="o">.</span><span class="n">s</span><span class="p">))</span>
    <span class="n">saveRDS</span><span class="p">(</span><span class="n">vhat</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[8]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># V estimate: &quot;mle&quot; method</span>
<span class="p">[</span><span class="n">vhat_mle</span><span class="p">]</span>
<span class="c1"># number of samples to use</span>
<span class="kn">parameter:</span> <span class="n">n_subset</span> <span class="o">=</span> <span class="mi">6000</span>
<span class="c1"># maximum number of iterations</span>
<span class="kn">parameter:</span> <span class="n">max_iter</span> <span class="o">=</span> <span class="mi">6</span>
<span class="kn">depends:</span> <span class="n">R_library</span><span class="p">(</span><span class="s2">&quot;mashr&quot;</span><span class="p">)</span>
<span class="kn">input:</span> <span class="n">data</span><span class="p">,</span> <span class="n">output_from</span><span class="p">(</span><span class="s1">&#39;prior&#39;</span><span class="p">)</span>
<span class="kn">output:</span> <span class="n">vhat_data</span>

<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;36h&#39;</span><span class="p">,</span> <span class="kp">trunk_size</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;4G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:bn}&#39;</span>
<span class="kn">R:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="kp">workdir</span> <span class="o">=</span> <span class="n">cwd</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stderr&quot;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stdout&quot;</span>
    <span class="n">library</span><span class="p">(</span><span class="n">mashr</span><span class="p">)</span>
    <span class="n">dat</span> <span class="o">=</span> <span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">})</span>
    <span class="c1"># choose random subset</span>
    <span class="nb">set</span><span class="o">.</span><span class="n">seed</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>
    <span class="n">random</span><span class="o">.</span><span class="n">subset</span> <span class="o">=</span> <span class="n">sample</span><span class="p">(</span><span class="mi">1</span><span class="p">:</span><span class="n">nrow</span><span class="p">(</span><span class="n">dat</span><span class="err">$</span><span class="n">random</span><span class="o">.</span><span class="n">b</span><span class="p">),</span> <span class="nb">min</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">n_subset</span><span class="p">},</span> <span class="n">nrow</span><span class="p">(</span><span class="n">dat</span><span class="err">$</span><span class="n">random</span><span class="o">.</span><span class="n">b</span><span class="p">)))</span>
    <span class="n">random</span><span class="o">.</span><span class="n">subset</span> <span class="o">=</span> <span class="n">mash_set_data</span><span class="p">(</span><span class="n">dat</span><span class="err">$</span><span class="n">random</span><span class="o">.</span><span class="n">b</span><span class="p">[</span><span class="n">random</span><span class="o">.</span><span class="n">subset</span><span class="p">,],</span> <span class="n">dat</span><span class="err">$</span><span class="n">random</span><span class="o">.</span><span class="n">s</span><span class="p">[</span><span class="n">random</span><span class="o">.</span><span class="n">subset</span><span class="p">,],</span> <span class="n">alpha</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="mi">1</span> <span class="k">if</span> <span class="n">effect_model</span> <span class="o">==</span> <span class="s1">&#39;EZ&#39;</span> <span class="k">else</span> <span class="mi">0</span><span class="p">})</span>
    <span class="c1"># estimate V mle</span>
    <span class="n">vhat</span> <span class="o">=</span> <span class="n">estimate_null_correlation</span><span class="p">(</span><span class="n">random</span><span class="o">.</span><span class="n">subset</span><span class="p">,</span> <span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">1</span><span class="p">]:</span><span class="n">r</span><span class="p">}),</span> <span class="n">max_iter</span> <span class="o">=</span> <span class="err">$</span><span class="p">{</span><span class="n">max_iter</span><span class="p">})</span>
    <span class="n">saveRDS</span><span class="p">(</span><span class="n">vhat</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Estimate each V separately via corshrink</span>
<span class="p">[</span><span class="n">vhat_corshrink_xcondition_1</span><span class="p">]</span>
<span class="c1"># Utility script</span>
<span class="kn">parameter:</span> <span class="n">util_script</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="s1">&#39;/project/mstephens/gtex/scripts/SumstatQuery.R&#39;</span><span class="p">)</span>
<span class="c1"># List of genes to analyze</span>
<span class="kn">parameter:</span> <span class="n">gene_list</span> <span class="o">=</span> <span class="n">path</span><span class="p">()</span>

<span class="n">fail_if</span><span class="p">(</span><span class="ow">not</span> <span class="n">gene_list</span><span class="o">.</span><span class="n">is_file</span><span class="p">(),</span> <span class="n">msg</span> <span class="o">=</span> <span class="s1">&#39;Please specify valid path for --gene-list&#39;</span><span class="p">)</span>
<span class="n">fail_if</span><span class="p">(</span><span class="ow">not</span> <span class="n">util_script</span><span class="o">.</span><span class="n">is_file</span><span class="p">()</span> <span class="ow">and</span> <span class="nb">len</span><span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">util_script</span><span class="p">)),</span> <span class="n">msg</span> <span class="o">=</span> <span class="s1">&#39;Please specify valid path for --util-script&#39;</span><span class="p">)</span>
<span class="n">genes</span> <span class="o">=</span> <span class="n">sort_uniq</span><span class="p">([</span><span class="n">x</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span><span class="o">.</span><span class="n">strip</span><span class="p">(</span><span class="s1">&#39;&quot;&#39;</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">open</span><span class="p">(</span><span class="n">f</span><span class="s1">&#39;{gene_list:a}&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">readlines</span><span class="p">()</span> <span class="k">if</span> <span class="ow">not</span> <span class="n">x</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s1">&#39;#&#39;</span><span class="p">)])</span>


<span class="kn">depends:</span> <span class="n">R_library</span><span class="p">(</span><span class="s2">&quot;CorShrink&quot;</span><span class="p">)</span>
<span class="kn">input:</span> <span class="n">data</span><span class="p">,</span> <span class="n">for_each</span> <span class="o">=</span> <span class="s1">&#39;genes&#39;</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s1">&#39;{vhat_data:n}/{_genes}.rds&#39;</span>

<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;3m&#39;</span><span class="p">,</span> <span class="kp">trunk_size</span> <span class="o">=</span> <span class="mi">500</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;3G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:bn}&#39;</span>
<span class="kn">R:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="kp">workdir</span> <span class="o">=</span> <span class="n">cwd</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stderr&quot;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stdout&quot;</span>
    <span class="n">source</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">util_script</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
    <span class="n">CorShrink_sum</span> <span class="o">=</span> <span class="n">function</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span> <span class="n">database</span><span class="p">,</span> <span class="n">z_thresh</span> <span class="o">=</span> <span class="mi">2</span><span class="p">){</span>
      <span class="k">print</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
      <span class="n">dat</span> <span class="o">&lt;-</span> <span class="n">GetSS</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span> <span class="n">database</span><span class="p">)</span>
      <span class="n">z</span> <span class="o">=</span> <span class="n">dat</span><span class="err">$</span><span class="s2">&quot;z-score&quot;</span>
      <span class="n">max_absz</span> <span class="o">=</span> <span class="nb">apply</span><span class="p">(</span><span class="nb">abs</span><span class="p">(</span><span class="n">z</span><span class="p">),</span> <span class="mi">1</span><span class="p">,</span> <span class="nb">max</span><span class="p">)</span>
      <span class="n">nullish</span> <span class="o">=</span> <span class="n">which</span><span class="p">(</span><span class="n">max_absz</span> <span class="o">&lt;</span> <span class="n">z_thresh</span><span class="p">)</span>
      <span class="c1"># if (length(nullish) &lt; ncol(z)) {</span>
        <span class="c1"># stop(&quot;not enough null data to estimate null correlation&quot;)</span>
      <span class="c1"># }</span>
      <span class="k">if</span> <span class="p">(</span><span class="n">length</span><span class="p">(</span><span class="n">nullish</span><span class="p">)</span> <span class="o">&lt;=</span> <span class="mi">1</span><span class="p">){</span>
        <span class="n">mat</span> <span class="o">=</span> <span class="n">diag</span><span class="p">(</span><span class="n">ncol</span><span class="p">(</span><span class="n">z</span><span class="p">))</span>
      <span class="p">}</span> <span class="k">else</span> <span class="p">{</span>
        <span class="n">nullish_z</span> <span class="o">=</span> <span class="n">z</span><span class="p">[</span><span class="n">nullish</span><span class="p">,</span> <span class="p">]</span>  
        <span class="n">mat</span> <span class="o">=</span> <span class="k">as</span><span class="o">.</span><span class="n">matrix</span><span class="p">(</span><span class="n">CorShrink</span><span class="p">::</span><span class="n">CorShrinkData</span><span class="p">(</span><span class="n">nullish_z</span><span class="p">,</span> <span class="n">ash</span><span class="o">.</span><span class="n">control</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">mixcompdist</span> <span class="o">=</span> <span class="s2">&quot;halfuniform&quot;</span><span class="p">))</span><span class="err">$</span><span class="n">cor</span><span class="p">)</span>
      <span class="p">}</span>
      <span class="k">return</span><span class="p">(</span><span class="n">mat</span><span class="p">)</span>
    <span class="p">}</span>
    <span class="n">V</span> <span class="o">=</span> <span class="n">Corshrink_sum</span><span class="p">(</span><span class="s2">&quot;${_genes}&quot;</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">data</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
    <span class="n">saveRDS</span><span class="p">(</span><span class="n">V</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>

<span class="c1"># Consolidate Vhat into one file</span>
<span class="p">[</span><span class="n">vhat_corshrink_xcondition_2</span><span class="p">,</span> <span class="n">vhat_simple_specific_2</span><span class="p">]</span>
<span class="kn">depends:</span> <span class="n">R_library</span><span class="p">(</span><span class="s2">&quot;parallel&quot;</span><span class="p">)</span>
<span class="c1"># List of genes to analyze</span>
<span class="kn">parameter:</span> <span class="n">gene_list</span> <span class="o">=</span> <span class="n">path</span><span class="p">()</span>

<span class="n">fail_if</span><span class="p">(</span><span class="ow">not</span> <span class="n">gene_list</span><span class="o">.</span><span class="n">is_file</span><span class="p">(),</span> <span class="n">msg</span> <span class="o">=</span> <span class="s1">&#39;Please specify valid path for --gene-list&#39;</span><span class="p">)</span>
<span class="n">genes</span> <span class="o">=</span> <span class="n">paths</span><span class="p">([</span><span class="n">x</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span><span class="o">.</span><span class="n">strip</span><span class="p">(</span><span class="s1">&#39;&quot;&#39;</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">open</span><span class="p">(</span><span class="n">f</span><span class="s1">&#39;{gene_list:a}&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">readlines</span><span class="p">()</span> <span class="k">if</span> <span class="ow">not</span> <span class="n">x</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s1">&#39;#&#39;</span><span class="p">)])</span>


<span class="kn">input:</span> <span class="n">group_by</span> <span class="o">=</span> <span class="s1">&#39;all&#39;</span>
<span class="kn">output:</span> <span class="n">vhat_data</span>

<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;1h&#39;</span><span class="p">,</span> <span class="kp">trunk_size</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;4G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:bn}&#39;</span>
<span class="kn">R:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="kp">workdir</span> <span class="o">=</span> <span class="n">cwd</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stderr&quot;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stdout&quot;</span>
    <span class="n">library</span><span class="p">(</span><span class="n">parallel</span><span class="p">)</span>
    <span class="n">files</span> <span class="o">=</span> <span class="n">sapply</span><span class="p">(</span><span class="n">c</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">genes</span><span class="p">:</span><span class="n">r</span><span class="p">,}),</span> <span class="n">function</span><span class="p">(</span><span class="n">g</span><span class="p">)</span> <span class="n">paste0</span><span class="p">(</span><span class="n">c</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">adr</span><span class="p">}),</span> <span class="s1">&#39;/&#39;</span><span class="p">,</span> <span class="n">g</span><span class="p">,</span> <span class="s1">&#39;.rds&#39;</span><span class="p">),</span> <span class="n">USE</span><span class="o">.</span><span class="n">NAMES</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span>
    <span class="n">V</span> <span class="o">=</span> <span class="n">mclapply</span><span class="p">(</span><span class="n">files</span><span class="p">,</span> <span class="n">function</span><span class="p">(</span><span class="n">i</span><span class="p">){</span> <span class="n">readRDS</span><span class="p">(</span><span class="n">i</span><span class="p">)</span> <span class="p">},</span> <span class="n">mc</span><span class="o">.</span><span class="kp">cores</span> <span class="o">=</span> <span class="mi">1</span><span class="p">)</span>
    <span class="n">R</span> <span class="o">=</span> <span class="n">dim</span><span class="p">(</span><span class="n">V</span><span class="p">[[</span><span class="mi">1</span><span class="p">]])[</span><span class="mi">1</span><span class="p">]</span>
    <span class="n">L</span> <span class="o">=</span> <span class="n">length</span><span class="p">(</span><span class="n">V</span><span class="p">)</span>
    <span class="n">V</span><span class="o">.</span><span class="n">array</span> <span class="o">=</span> <span class="n">array</span><span class="p">(</span><span class="k">as</span><span class="o">.</span><span class="n">numeric</span><span class="p">(</span><span class="n">unlist</span><span class="p">(</span><span class="n">V</span><span class="p">)),</span> <span class="n">dim</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="n">R</span><span class="p">,</span> <span class="n">R</span><span class="p">,</span> <span class="n">L</span><span class="p">))</span>
    <span class="n">saveRDS</span><span class="p">(</span><span class="n">V</span><span class="o">.</span><span class="n">array</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">ar</span><span class="p">})</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Estimate each V separately via &quot;simple&quot; method</span>
<span class="p">[</span><span class="n">vhat_simple_specific_1</span><span class="p">]</span>
<span class="c1"># Utility script</span>
<span class="kn">parameter:</span> <span class="n">util_script</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="s1">&#39;/project/mstephens/gtex/scripts/SumstatQuery.R&#39;</span><span class="p">)</span>
<span class="c1"># List of genes to analyze</span>
<span class="kn">parameter:</span> <span class="n">gene_list</span> <span class="o">=</span> <span class="n">path</span><span class="p">()</span>

<span class="n">fail_if</span><span class="p">(</span><span class="ow">not</span> <span class="n">gene_list</span><span class="o">.</span><span class="n">is_file</span><span class="p">(),</span> <span class="n">msg</span> <span class="o">=</span> <span class="s1">&#39;Please specify valid path for --gene-list&#39;</span><span class="p">)</span>
<span class="n">fail_if</span><span class="p">(</span><span class="ow">not</span> <span class="n">util_script</span><span class="o">.</span><span class="n">is_file</span><span class="p">()</span> <span class="ow">and</span> <span class="nb">len</span><span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">util_script</span><span class="p">)),</span> <span class="n">msg</span> <span class="o">=</span> <span class="s1">&#39;Please specify valid path for --util-script&#39;</span><span class="p">)</span>
<span class="n">genes</span> <span class="o">=</span> <span class="n">sort_uniq</span><span class="p">([</span><span class="n">x</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span><span class="o">.</span><span class="n">strip</span><span class="p">(</span><span class="s1">&#39;&quot;&#39;</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">open</span><span class="p">(</span><span class="n">f</span><span class="s1">&#39;{gene_list:a}&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">readlines</span><span class="p">()</span> <span class="k">if</span> <span class="ow">not</span> <span class="n">x</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s1">&#39;#&#39;</span><span class="p">)])</span>

<span class="kn">depends:</span> <span class="n">R_library</span><span class="p">(</span><span class="s2">&quot;Matrix&quot;</span><span class="p">)</span>
<span class="kn">input:</span> <span class="n">data</span><span class="p">,</span> <span class="n">for_each</span> <span class="o">=</span> <span class="s1">&#39;genes&#39;</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s1">&#39;{vhat_data:n}/{_genes}.rds&#39;</span>

<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;1m&#39;</span><span class="p">,</span> <span class="kp">trunk_size</span> <span class="o">=</span> <span class="mi">500</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;3G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:bn}&#39;</span>
<span class="kn">R:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="kp">workdir</span> <span class="o">=</span> <span class="n">cwd</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stderr&quot;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stdout&quot;</span>
    <span class="n">source</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">util_script</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
    <span class="n">simple_V</span> <span class="o">=</span> <span class="n">function</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span> <span class="n">database</span><span class="p">,</span> <span class="n">z_thresh</span> <span class="o">=</span> <span class="mi">2</span><span class="p">){</span>
      <span class="k">print</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
      <span class="n">dat</span> <span class="o">&lt;-</span> <span class="n">GetSS</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span> <span class="n">database</span><span class="p">)</span>
      <span class="n">z</span> <span class="o">=</span> <span class="n">dat</span><span class="err">$</span><span class="s2">&quot;z-score&quot;</span>
      <span class="n">max_absz</span> <span class="o">=</span> <span class="nb">apply</span><span class="p">(</span><span class="nb">abs</span><span class="p">(</span><span class="n">z</span><span class="p">),</span> <span class="mi">1</span><span class="p">,</span> <span class="nb">max</span><span class="p">)</span>
      <span class="n">nullish</span> <span class="o">=</span> <span class="n">which</span><span class="p">(</span><span class="n">max_absz</span> <span class="o">&lt;</span> <span class="n">z_thresh</span><span class="p">)</span>
      <span class="c1"># if (length(nullish) &lt; ncol(z)) {</span>
        <span class="c1"># stop(&quot;not enough null data to estimate null correlation&quot;)</span>
      <span class="c1"># }</span>
      <span class="k">if</span> <span class="p">(</span><span class="n">length</span><span class="p">(</span><span class="n">nullish</span><span class="p">)</span> <span class="o">&lt;=</span> <span class="mi">1</span><span class="p">){</span>
        <span class="n">mat</span> <span class="o">=</span> <span class="n">diag</span><span class="p">(</span><span class="n">ncol</span><span class="p">(</span><span class="n">z</span><span class="p">))</span>
      <span class="p">}</span> <span class="k">else</span> <span class="p">{</span>
        <span class="n">nullish_z</span> <span class="o">=</span> <span class="n">z</span><span class="p">[</span><span class="n">nullish</span><span class="p">,</span> <span class="p">]</span>
        <span class="n">mat</span> <span class="o">=</span> <span class="k">as</span><span class="o">.</span><span class="n">matrix</span><span class="p">(</span><span class="n">Matrix</span><span class="p">::</span><span class="n">nearPD</span><span class="p">(</span><span class="k">as</span><span class="o">.</span><span class="n">matrix</span><span class="p">(</span><span class="n">cov</span><span class="p">(</span><span class="n">nullish_z</span><span class="p">)),</span> <span class="n">conv</span><span class="o">.</span><span class="n">tol</span><span class="o">=</span><span class="mf">1e-06</span><span class="p">,</span> <span class="n">doSym</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">corr</span><span class="o">=</span><span class="n">TRUE</span><span class="p">)</span><span class="err">$</span><span class="n">mat</span><span class="p">)</span>
      <span class="p">}</span>
      <span class="k">return</span><span class="p">(</span><span class="n">mat</span><span class="p">)</span>
    <span class="p">}</span>
    <span class="n">V</span> <span class="o">=</span> <span class="n">simple_V</span><span class="p">(</span><span class="s2">&quot;${_genes}&quot;</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">data</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
    <span class="n">saveRDS</span><span class="p">(</span><span class="n">V</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="mashr-mixture-model-fitting"><code>mashr</code> mixture model fitting<a class="anchor-link" href="#mashr-mixture-model-fitting">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[9]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Fit MASH mixture model (time estimate: &lt;15min for 70K by 49 matrix)</span>
<span class="p">[</span><span class="n">mash_1</span><span class="p">]</span>
<span class="kn">depends:</span> <span class="n">R_library</span><span class="p">(</span><span class="s2">&quot;mashr&quot;</span><span class="p">)</span>
<span class="kn">input:</span> <span class="n">data</span><span class="p">,</span> <span class="n">output_from</span><span class="p">(</span><span class="s2">&quot;prior&quot;</span><span class="p">),</span> <span class="n">output_from</span><span class="p">(</span><span class="n">f</span><span class="s2">&quot;vhat_{vhat}&quot;</span><span class="p">)</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s2">&quot;{_input[-1]:n}.mash_model.rds&quot;</span>

<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;36h&#39;</span><span class="p">,</span> <span class="kp">trunk_size</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;4G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:bn}&#39;</span>
<span class="kn">R:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="kp">workdir</span> <span class="o">=</span> <span class="n">cwd</span><span class="p">,</span> <span class="kp">env</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;MOSEKLM_LICENSE_FILE&#39;</span><span class="p">:</span> <span class="nb">str</span><span class="p">(</span><span class="n">mosek_license</span><span class="p">)},</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stderr&quot;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stdout&quot;</span>
    <span class="n">library</span><span class="p">(</span><span class="n">mashr</span><span class="p">)</span>
    <span class="n">dat</span> <span class="o">=</span> <span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">})</span>
    <span class="n">vhat</span> <span class="o">=</span> <span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">2</span><span class="p">]:</span><span class="n">r</span><span class="p">})</span>
    <span class="n">mash_data</span> <span class="o">=</span> <span class="n">mash_set_data</span><span class="p">(</span><span class="n">dat</span><span class="err">$</span><span class="n">random</span><span class="o">.</span><span class="n">b</span><span class="p">,</span> <span class="n">Shat</span><span class="o">=</span><span class="n">dat</span><span class="err">$</span><span class="n">random</span><span class="o">.</span><span class="n">s</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="mi">1</span> <span class="k">if</span> <span class="n">effect_model</span> <span class="o">==</span> <span class="s1">&#39;EZ&#39;</span> <span class="k">else</span> <span class="mi">0</span><span class="p">},</span> <span class="n">V</span><span class="o">=</span><span class="n">vhat</span><span class="p">)</span>
    <span class="n">saveRDS</span><span class="p">(</span><span class="n">mash</span><span class="p">(</span><span class="n">mash_data</span><span class="p">,</span> <span class="n">Ulist</span> <span class="o">=</span> <span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">1</span><span class="p">]:</span><span class="n">r</span><span class="p">}),</span> <span class="n">optmethod</span> <span class="o">=</span> <span class="s2">&quot;${optmethod}&quot;</span><span class="p">,</span> <span class="n">outputlevel</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="n">algorithm</span><span class="o">.</span><span class="n">version</span> <span class="o">=</span> <span class="s2">&quot;${implementation}&quot;</span><span class="p">),</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Optional-posterior-computations">Optional posterior computations<a class="anchor-link" href="#Optional-posterior-computations">&#182;</a></h4><p>Additionally provide posterior for the "strong" set in MASH input data.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[10]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Compute posterior for the &quot;strong&quot; set of data as in Urbut et al 2017.</span>
<span class="c1"># This is optional because most of the time we want to apply the </span>
<span class="c1"># MASH model learned on much larger data-set.</span>
<span class="p">[</span><span class="n">mash_2</span><span class="p">]</span>
<span class="c1"># default to True; use --no-compute-posterior to disable this</span>
<span class="kn">parameter:</span> <span class="n">compute_posterior</span> <span class="o">=</span> <span class="bp">True</span>
<span class="c1"># input Vhat file for the batch of posterior data</span>
<span class="kn">parameter:</span> <span class="n">posterior_vhat_file</span> <span class="o">=</span> <span class="n">path</span><span class="p">()</span>
<span class="n">skip_if</span><span class="p">(</span><span class="ow">not</span> <span class="n">compute_posterior</span><span class="p">)</span>
<span class="n">fail_if</span><span class="p">(</span><span class="ow">not</span> <span class="n">posterior_vhat_file</span><span class="o">.</span><span class="n">is_file</span><span class="p">()</span> <span class="ow">and</span> <span class="nb">len</span><span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">posterior_vhat_file</span><span class="p">))</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">,</span> <span class="n">msg</span> <span class="o">=</span> <span class="s1">&#39;Cannot find specified --posterior-vhat-file. Please provide the correct path for it!&#39;</span><span class="p">)</span>
<span class="kn">depends:</span> <span class="n">R_library</span><span class="p">(</span><span class="s2">&quot;mashr&quot;</span><span class="p">)</span>
<span class="kn">input:</span> <span class="n">data</span><span class="p">,</span> <span class="n">posterior_vhat_file</span> <span class="k">if</span> <span class="n">posterior_vhat_file</span><span class="o">.</span><span class="n">is_file</span><span class="p">()</span> <span class="k">else</span> <span class="n">output_from</span><span class="p">(</span><span class="n">f</span><span class="s2">&quot;vhat_{vhat}&quot;</span><span class="p">),</span> <span class="n">output_from</span><span class="p">(</span><span class="o">-</span><span class="mi">1</span><span class="p">)</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s1">&#39;{posterior_vhat_file:n}.posterior.rds&#39;</span> <span class="k">if</span> <span class="n">posterior_vhat_file</span><span class="o">.</span><span class="n">is_file</span><span class="p">()</span> <span class="k">else</span> <span class="n">f</span><span class="s2">&quot;{_input[-1]:nn}.posterior.rds&quot;</span>

<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;36h&#39;</span><span class="p">,</span> <span class="kp">trunk_size</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;4G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:bn}&#39;</span>
<span class="kn">R:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="kp">workdir</span> <span class="o">=</span> <span class="n">cwd</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stderr&quot;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stdout&quot;</span>
    <span class="n">library</span><span class="p">(</span><span class="n">mashr</span><span class="p">)</span>
    <span class="n">dat</span> <span class="o">=</span> <span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">})</span>
    <span class="n">vhat</span> <span class="o">=</span> <span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">1</span><span class="p">]:</span><span class="n">r</span><span class="p">})</span>
    <span class="n">mash_data</span> <span class="o">=</span> <span class="n">mash_set_data</span><span class="p">(</span><span class="n">dat</span><span class="err">$</span><span class="n">strong</span><span class="o">.</span><span class="n">b</span><span class="p">,</span> <span class="n">Shat</span><span class="o">=</span><span class="n">dat</span><span class="err">$</span><span class="n">strong</span><span class="o">.</span><span class="n">s</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="mi">1</span> <span class="k">if</span> <span class="n">effect_model</span> <span class="o">==</span> <span class="s1">&#39;EZ&#39;</span> <span class="k">else</span> <span class="mi">0</span><span class="p">},</span> <span class="n">V</span><span class="o">=</span><span class="n">vhat</span><span class="p">)</span>
    <span class="n">mash_model</span> <span class="o">=</span> <span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">2</span><span class="p">]:</span><span class="n">ar</span><span class="p">})</span>
    <span class="n">saveRDS</span><span class="p">(</span><span class="n">mash_compute_posterior_matrices</span><span class="p">(</span><span class="n">mash_model</span><span class="p">,</span> <span class="n">mash_data</span><span class="p">,</span> <span class="n">algorithm</span><span class="o">.</span><span class="n">version</span> <span class="o">=</span> <span class="s2">&quot;${implementation}&quot;</span><span class="p">),</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Compute-MASH-posteriors">Compute MASH posteriors<a class="anchor-link" href="#Compute-MASH-posteriors">&#182;</a></h3><p>In the GTEx V6 paper we assumed one eQTL per gene and applied the model learned above to those SNPs. Under that assumption, the input data for posterior calculation will be the <code>dat$strong.*</code> matrices.
It is a fairly straightforward procedure as shown in <a href="https://stephenslab.github.io/mashr/articles/eQTL_outline.html">this vignette</a>.</p>
<p>But it is often more interesting to apply MASH to given list of eQTLs, eg, from those from fine-mapping results. In GTEx V8 analysis we obtain such gene-SNP pairs from DAP-G fine-mapping analysis. See <a href="https://gaow.github.io/mnm-gtex-v8/analysis/Independent_eQTL_Results.html">this notebook</a> for how the input data is prepared. The workflow below takes a number of input chunks (each chunk is a list of matrices <code>dat$Bhat</code> and <code>dat$Shat</code>) 
and computes posterior for each chunk. It is therefore suited for running in parallel posterior computation for all gene-SNP pairs, if input data chunks are provided.</p>

<pre><code>JOB_OPT="-c midway2.yml -q midway2"
DATA_DIR=/project/compbio/GTEx_eQTL/independent_eQTL
sos run workflows/mashr_flashr_workflow.ipynb posterior \
    $JOB_OPT \
    --posterior-input $DATA_DIR/DAPG_pip_gt_0.01-AllTissues/DAPG_pip_gt_0.01-AllTissues.*.rds \
                      $DATA_DIR/ConditionalAnalysis_AllTissues/ConditionalAnalysis_AllTissues.*.rds</code></pre>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[11]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Apply posterior calculations</span>
<span class="p">[</span><span class="n">posterior</span><span class="p">]</span>
<span class="kn">parameter:</span> <span class="n">mash_model</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="n">f</span><span class="s2">&quot;{vhat_data:n}.mash_model.rds&quot;</span><span class="p">)</span>
<span class="kn">parameter:</span> <span class="n">posterior_input</span> <span class="o">=</span> <span class="n">paths</span><span class="p">()</span>
<span class="kn">parameter:</span> <span class="n">posterior_vhat_files</span> <span class="o">=</span> <span class="n">paths</span><span class="p">()</span>
<span class="c1"># eg, if data is saved in R list as data$strong, then</span>
<span class="c1"># when you specify `--data-table-name strong` it will read the data as</span>
<span class="c1"># readRDS(&#39;{_input:r}&#39;)$strong</span>
<span class="kn">parameter:</span> <span class="n">data_table_name</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>
<span class="kn">parameter:</span> <span class="n">bhat_table_name</span> <span class="o">=</span> <span class="s1">&#39;Bhat&#39;</span>
<span class="kn">parameter:</span> <span class="n">shat_table_name</span> <span class="o">=</span> <span class="s1">&#39;Shat&#39;</span>

<span class="n">skip_if</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">posterior_input</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">,</span> <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;No posterior input data to compute on. Please specify it using --posterior-input.&quot;</span><span class="p">)</span>
<span class="n">fail_if</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">posterior_vhat_files</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span> <span class="ow">and</span> <span class="nb">len</span><span class="p">(</span><span class="n">posterior_vhat_files</span><span class="p">)</span> <span class="o">!=</span> <span class="nb">len</span><span class="p">(</span><span class="n">posterior_input</span><span class="p">),</span> <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;length of --posterior-input and --posterior-vhat-files do not agree.&quot;</span><span class="p">)</span>
<span class="k">for</span> <span class="n">p</span> <span class="ow">in</span> <span class="n">posterior_input</span><span class="p">:</span>
    <span class="n">fail_if</span><span class="p">(</span><span class="ow">not</span> <span class="n">p</span><span class="o">.</span><span class="n">is_file</span><span class="p">(),</span> <span class="n">msg</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;Cannot find posterior input file ``{p}``&#39;</span><span class="p">)</span>

<span class="kn">depends:</span> <span class="n">R_library</span><span class="p">(</span><span class="s2">&quot;mashr&quot;</span><span class="p">)</span>
<span class="kn">input:</span> <span class="n">posterior_input</span><span class="p">,</span> <span class="n">group_by</span> <span class="o">=</span> <span class="mi">1</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s2">&quot;{_input:n}.posterior.rds&quot;</span>
    
<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;20h&#39;</span><span class="p">,</span> <span class="kp">trunk_size</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;20G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:bn}&#39;</span>
<span class="kn">R:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="kp">workdir</span> <span class="o">=</span> <span class="n">cwd</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stderr&quot;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s2">&quot;{_output:n}.stdout&quot;</span>
    <span class="n">library</span><span class="p">(</span><span class="n">mashr</span><span class="p">)</span>
    <span class="n">data</span> <span class="o">=</span> <span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">:</span><span class="n">r</span><span class="p">})</span><span class="err">$</span><span class="p">{(</span><span class="s1">&#39;$&#39;</span> <span class="o">+</span> <span class="n">data_table_name</span><span class="p">)</span> <span class="k">if</span> <span class="n">data_table_name</span> <span class="k">else</span> <span class="s1">&#39;&#39;</span><span class="p">}</span>
    <span class="n">vhat</span> <span class="o">=</span> <span class="n">readRDS</span><span class="p">(</span><span class="s2">&quot;${vhat_data if len(posterior_vhat_files) == 0 else posterior_vhat_files[_index]}&quot;</span><span class="p">)</span>
    <span class="n">mash_data</span> <span class="o">=</span> <span class="n">mash_set_data</span><span class="p">(</span><span class="n">data</span><span class="err">$$</span><span class="p">{</span><span class="n">bhat_table_name</span><span class="p">},</span> <span class="n">Shat</span><span class="o">=</span><span class="n">data</span><span class="err">$$</span><span class="p">{</span><span class="n">shat_table_name</span><span class="p">},</span> <span class="n">alpha</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="mi">1</span> <span class="k">if</span> <span class="n">effect_model</span> <span class="o">==</span> <span class="s1">&#39;EZ&#39;</span> <span class="k">else</span> <span class="mi">0</span><span class="p">},</span> <span class="n">V</span><span class="o">=</span><span class="n">vhat</span><span class="p">)</span>
    <span class="n">saveRDS</span><span class="p">(</span><span class="n">mash_compute_posterior_matrices</span><span class="p">(</span><span class="n">readRDS</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">mash_model</span><span class="p">:</span><span class="n">r</span><span class="p">}),</span> <span class="n">mash_data</span><span class="p">,</span> <span class="n">algorithm</span> <span class="o">=</span> <span class="s2">&quot;${implementation}&quot;</span><span class="p">),</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">r</span><span class="p">})</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Posterior-results">Posterior results<a class="anchor-link" href="#Posterior-results">&#182;</a></h4>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<ol>
<li>The outcome of the <code>[posterior]</code> step should produce a number of serialized R objects <code>*.batch_*.posterior.rds</code> (can be loaded to R via <code>readRDS()</code>) -- I chopped data to batches to take advantage of computing in multiple cluster nodes. It should be self-explanary but please let me know otherwise.</li>
<li>Other posterior related files are:<ol>
<li><code>*.batch_*.yaml</code>: gene-SNP pairs of interest, identified elsewhere (eg. fine-mapping analysis). </li>
<li>The corresponding univariate analysis summary statistics for gene-SNPs from <code>*.batch_*.yaml</code> are extracted and saved to <code>*.batch_*.rds</code>, creating input to the <code>[posterior]</code> step.</li>
<li>Note the <code>*.batch_*.stdout</code> file documents some SNPs found in fine-mapping results but not found in the original <code>fastqtl</code> output.</li>
</ol>
</li>
</ol>

</div>
</div>
</div>
<hr>
&copy 2018 Gao Wang, University of Chicago
<p><small>Exported from <a href="http://github.com/stephenslab/gtex-eqtls/blob/4e7b5daea46f00152b4f930c318a2f65aedda5d7/analysis/mashr_flashr_workflow.ipynb"><code>analysis/mashr_flashr_workflow.ipynb</code></a> committed by Gao Wang on Tue Feb 2 14:25:48 2021 <a href="http://github.com/stephenslab/gtex-eqtls/commit/4e7b5daea46f00152b4f930c318a2f65aedda5d7">revision 2, 4e7b5da</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
<p><small>Exported from <a href="http://github.com/stephenslab/gtex-eqtls/blob/4e7b5daea46f00152b4f930c318a2f65aedda5d7/analysis/mashr_flashr_workflow.ipynb"><code>analysis/mashr_flashr_workflow.ipynb</code></a> committed by Gao Wang on Tue Feb 2 14:25:48 2021 <a href="http://github.com/stephenslab/gtex-eqtls/commit/4e7b5daea46f00152b4f930c318a2f65aedda5d7">revision 2, 4e7b5da</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
<p><small>Exported from <a href="http://github.com/stephenslab/gtex-eqtls/blob/c5fe213315f2a5667a5fc5a08877046fd5f62cf0/analysis/mashr_flashr_workflow.ipynb"><code>analysis/mashr_flashr_workflow.ipynb</code></a> committed by Gao Wang on Tue Feb 2 14:11:23 2021 <a href="http://github.com/stephenslab/gtex-eqtls/commit/c5fe213315f2a5667a5fc5a08877046fd5f62cf0">revision 1, c5fe213</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
<p><small>Exported from <a href="http://github.com/gaow/mnm-gtex-v8/blob/7c827e2dde99d833e84a696e78094f69804ad162/analysis/mashr_flashr_workflow.ipynb"><code>analysis/mashr_flashr_workflow.ipynb</code></a> committed by Gao Wang on Sat Sep 28 13:25:34 2019 <a href="http://github.com/gaow/mnm-gtex-v8/commit/7c827e2dde99d833e84a696e78094f69804ad162">revision 11, 7c827e2</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
