{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Meetings"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 20171211 GTEx fine mapping workgroup meeting\n",
    "Phone conference with GTEx members from many cities ... Andrew presented.\n",
    "\n",
    "Software to apply:\n",
    "- Farhad Caviar: PIP on all tissues, 3 days\n",
    "- Andrew Caveman: PIP on all tissues, but whole blood first, a couple of days; all tissues 3 weeks\n",
    "- William DAP-G: a couple of days per tissue. A couple of tissues to start from.\n",
    "  - PIP\n",
    "  - LD clusters\n",
    "- Possibly 1 or 2 more software\n",
    "- Results for all tissues will be available mid-Jan.\n",
    "\n",
    "Data:\n",
    "- Francois: VCF are almost final, though missingness are causing confusion (but in short we can mean-impute them). Should be finalized mid-week this week\n",
    "- Andrew: ChIP-seq benchmarks (an annotation list) to be shared. The data comes from roadmap epigenomics and will be lifted over to V9 data.\n",
    "- Tuuli: Has annotation for post transcriptional modification\n",
    "- William: Has annotation for TFBS.\n",
    "\n",
    "Simulation data: \n",
    "- Andrew: not necessary. or at least should not be too close to the way any methods work. eg, no one-eqtl per gene type of setup.\n",
    "- William: is necessary. no need to worry about generating from certain models. Simulation based on simple linear model is generic enough.\n",
    "- Eleazar: Simulation results not interesting, compared to ChIP-seq / other annotation benchmarks.\n",
    "\n",
    "Exploration then benchmarking, or benchmarking right now?\n",
    "- Blood results before Christmas, so that benchmarking can be put together.\n",
    "\n",
    "Benchmarking:\n",
    "- Input: data and annotation. Annotation can be used for tests, or for validation\n",
    "  - Caveman now taks pre-computed annotations\n",
    "  - Some type of Enrichment summary\n",
    "- Output: scores?\n",
    "  - Gao has to figure out these\n",
    "  - PIP, lfdr? at variant vs. gene level? how about LD cluster based methods?\n",
    "- Blood results ready in a week to start putting together benchmarks. \n",
    "- All tissues by mid-Jan\n",
    "  - data availability\n",
    "\n",
    "Extension of methods:\n",
    "- cell-specific eQTLs\n",
    "- splicing eQTLs\n",
    "- age dependent QTLs\n",
    "- co-localisation\n",
    "  - Haky: GWAS group is also interested\n",
    "- multiple traits\n",
    "  - Tuuli: can anyone put together a quick&dirty M&M analysis? \n",
    "    - Tuuli's group is developing something\n",
    "  - Andrew: pairwise tissue analysis\n",
    "  - William: can look at a few tissues\n",
    "\n",
    "\n",
    "## 20171002 GTEx fine mapping meeting\n",
    "Goal: methods do not repeat each-other; should build on each other.\n",
    "\n",
    "### What is fine-mapping\n",
    "* probability estimates of each variant being causal\n",
    "    * how to verify it? experimentally?\n",
    "* Should use 1MB region?\n",
    "    * 100KB should not lose a lot.\n",
    "\n",
    "### Fine-mapping methods\n",
    "Analyse/benchmark one data-set using all methods. The best tissue might be Thyroid to test it out.\n",
    "* people are interested in it.\n",
    "* we can report results in a few weeks\n",
    "\n",
    "Methods out there (in the conference call):\n",
    "* CaVEMaN, DAP, CAVIAR\n",
    "* Can provide these results in a few weeks\n",
    "    * 2MB radius\n",
    "    * on Thyroid data\n",
    "    * channel in Slack for this topic\n",
    "\n",
    "General flavors\n",
    "* Enrichment of eQTLs giving annotations: \n",
    "    * @william: Same model but various ways to estimate priors. Open to different prior options. Also see concensus results from different methods is also important.\n",
    "    * But measuring consensus is difficult. Correlation is high but individual level there are problems.\n",
    "* Conditional analysis: soon to be uploaded to `exchange`\n",
    "* Additional type of integrated analysis\n",
    "    * @eric\n",
    "    \n",
    "Provide fine-mapping results for the community\n",
    "\n",
    "### cell-specific QTL    \n",
    "* Fine-mapping for **cell-specific** QTL: cell-type deconvolution to get proportion of cells.\n",
    "    * interaction terms / interaction testing?\n",
    "\n",
    "Provide fine-mapping results for the community\n",
    "\n",
    "### Splicing QTLs\n",
    "Can we fine-map splice QTLs?\n",
    "\n",
    "### Interest in fine-mapping companion paper?\n",
    "People are interested. There is also co-localization working group, but not progressing much.\n",
    "\n",
    "### Interest in multi-tissue fine-mapping?\n",
    "It is important but computationally hard. William is the only one who positively responed, but he only agree to do a few tissues.\n",
    "\n",
    "### Other related stuff\n",
    "* Sharing of results"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SoS",
   "language": "sos",
   "name": "sos"
  },
  "language_info": {
   "codemirror_mode": "sos",
   "file_extension": ".sos",
   "mimetype": "text/x-sos",
   "name": "sos",
   "nbconvert_exporter": "sos_notebook.converter.SoS_Exporter",
   "pygments_lexer": "sos"
  },
  "sos": {
   "default_kernel": "SoS",
   "kernels": [
    [
     "Python3",
     "python3",
     "Python3",
     "#FFE771"
    ],
    [
     "R",
     "ir",
     "R",
     "#DCDCDA"
    ],
    [
     "SoS",
     "sos",
     "",
     ""
    ]
   ],
   "panel": {
    "displayed": true,
    "height": 0,
    "style": "side"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
