{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Profile and improve data conversion pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The [data conversion pipeline](https://github.com/stephenslab/gtex-eqtls/blob/master/analysis/fastqtl_to_mash.ipynb) was initially written in 2015 to deal with GTEx V6 summary statistics data processing for cis-eQTL mapping. But when working with sQTL data of 10 times large the line counts, it is no longer suitable. \n",
    "\n",
    "There are a few things one might be able to do but I think the easiest is to profile the code and see if there are some cheap improvement to be made; because I do suspect a couple of places the code is slow."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## The test data\n",
    "\n",
    "I get a small test file:\n",
    "\n",
    "```\n",
    "[MW] zcat allpair-at-intron-level.Testis.txt.gz | head -100000 | gzip > test.txt.gz\n",
    "[MW] ll test.txt.gz \n",
    "-rw-rw-r-- 1 gaow pi-mstephens 2.4M Sep 28 12:27 test.txt.gz\n",
    "```\n",
    "then the gene list, `gtex_v8_sqtl_eur_only.genes_list` for 19K gene names."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## The code to profile\n",
    "\n",
    "I copy codes from the pipeline and put it here."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing test.py\n"
     ]
    }
   ],
   "source": [
    "%%writefile test.py\n",
    "cols = [7,6,8]\n",
    "import numpy as np\n",
    "import copy\n",
    "\n",
    "class SSData:\n",
    "    def __init__(self, header = False):\n",
    "        self.data = {'buffer':{'data':[], 'rownames':[]}, 'output':{}}\n",
    "        self.header = header\n",
    "        self.previous_name = self.current_name = None\n",
    "        self.count = -1\n",
    "\n",
    "    @profile\n",
    "    def parse(self, line, ensg_version = 0, include = None):\n",
    "        # input line is snp, gene, beta, t, pval and optionally se(bhat)\n",
    "        if not line:\n",
    "            self.__reset()\n",
    "            self.current_name = None\n",
    "            return 1\n",
    "        if isinstance(line, bytes):\n",
    "            line = line.decode()\n",
    "        line = line.strip().split()\n",
    "        self.count += 1\n",
    "        if self.header and self.count == 0:\n",
    "            return 0\n",
    "        # the line is not long enough ... there is format issues\n",
    "        if len(line) < max([cols[0], cols[1], cols[2]]):\n",
    "            return -1\n",
    "        #\n",
    "        line[0] = line[0].strip('\"')\n",
    "        if ensg_version == 0:\n",
    "            line[0] = line[0].split('.')[0]\n",
    "        #\n",
    "        if include is not None and line[0] not in include:\n",
    "            return -1\n",
    "        if self.previous_name is None:\n",
    "            self.previous_name = line[0]\n",
    "        self.current_name = line[0]\n",
    "        if self.current_name != self.previous_name:\n",
    "            self.__reset()\n",
    "        if cols[2] == 0:\n",
    "            # the two numbers are bhat and p-value, we need to compute se(bhat)\n",
    "            self.data['buffer']['data'].append([line[cols[0]-1], str(get_se(float(line[cols[0]-1]), float(line[cols[1]-1]))), line[cols[1]-1]])\n",
    "        else:\n",
    "            self.data['buffer']['data'].append([line[cols[0]-1], line[cols[1]-1], line[cols[2]-1]])\n",
    "        self.data['buffer']['rownames'].append(self.__format_variant_id(line[1]))\n",
    "        return 0\n",
    "\n",
    "    @staticmethod\n",
    "    def __format_variant_id(value):\n",
    "        value = value.strip('\"').split('_')\n",
    "        if len(value) > 4:\n",
    "            # keep it chr, pos, ref, alt\n",
    "            value = value[:4]\n",
    "        if value[0].startswith('chr'):\n",
    "            value[0] = value[0][3:]\n",
    "        return '_'.join(value)\n",
    "\n",
    "    def __reset(self):\n",
    "        self.data['buffer']['data'] = np.array(self.data['buffer']['data'], dtype = float)\n",
    "        self.data['buffer']['rownames'] = np.array(self.data['buffer']['rownames'])\n",
    "        self.data['buffer']['colnames'] = np.array(['beta','se','pval'])\n",
    "        self.data['output'] = copy.deepcopy(self.data['buffer'])\n",
    "        self.data['buffer'] = {'data':[], 'rownames':[]}\n",
    "\n",
    "    def dump(self):\n",
    "        return self.data['output']\n",
    "\n",
    "@profile\n",
    "def test(sumstat_file, gene_list, keep_ensg_version = 0):\n",
    "    import warnings\n",
    "    import gzip, os\n",
    "    gene_names = [x.strip() for x in open(gene_list).readlines() if x.strip()]\n",
    "    if keep_ensg_version == 0 and gene_names is not None:\n",
    "        gene_names = [os.path.splitext(x)[0] for x in gene_names]\n",
    "\n",
    "    ssp = SSData(header = True)\n",
    "    group_counts = 0\n",
    "    with gzip.open(sumstat_file) as f:\n",
    "        while True:\n",
    "            line = f.readline()\n",
    "            quit = ssp.parse(line, keep_ensg_version, include = gene_names)\n",
    "            if quit == -1:\n",
    "                continue\n",
    "            if ssp.current_name != ssp.previous_name:\n",
    "                group_counts += 1\n",
    "                ssp.previous_name = ssp.current_name\n",
    "            if quit:\n",
    "                break\n",
    "test('test.txt.gz', 'gtex_v8_sqtl_eur_only.genes_list')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wrote profile results to test.py.lprof\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "kernprof -l test.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Timer unit: 1e-06 s\n",
      "\n",
      "Total time: 31.5782 s\n",
      "File: test.py\n",
      "Function: parse at line 12\n",
      "\n",
      "Line #      Hits         Time  Per Hit   % Time  Line Contents\n",
      "==============================================================\n",
      "    12                                               @profile\n",
      "    13                                               def parse(self, line, ensg_version = 0, include = None):\n",
      "    14                                                   # input line is snp, gene, beta, t, pval and optionally se(bhat)\n",
      "    15    100001      64018.0      0.6      0.2          if not line:\n",
      "    16         1       3179.0   3179.0      0.0              self.__reset()\n",
      "    17         1          2.0      2.0      0.0              self.current_name = None\n",
      "    18         1          0.0      0.0      0.0              return 1\n",
      "    19    100000      82349.0      0.8      0.3          if isinstance(line, bytes):\n",
      "    20    100000      95067.0      1.0      0.3              line = line.decode()\n",
      "    21    100000     150807.0      1.5      0.5          line = line.strip().split()\n",
      "    22    100000      92677.0      0.9      0.3          self.count += 1\n",
      "    23    100000      73105.0      0.7      0.2          if self.header and self.count == 0:\n",
      "    24         1          1.0      1.0      0.0              return 0\n",
      "    25                                                   # the line is not long enough ... there is format issues\n",
      "    26     99999     151929.0      1.5      0.5          if len(line) < max([cols[0], cols[1], cols[2]]):\n",
      "    27                                                       return -1\n",
      "    28                                                   #\n",
      "    29     99999      97352.0      1.0      0.3          line[0] = line[0].strip('\"')\n",
      "    30     99999      62109.0      0.6      0.2          if ensg_version == 0:\n",
      "    31     99999     100999.0      1.0      0.3              line[0] = line[0].split('.')[0]\n",
      "    32                                                   #\n",
      "    33     99999   30469844.0    304.7     96.5          if include is not None and line[0] not in include:\n",
      "    34     94391      68591.0      0.7      0.2              return -1\n",
      "    35      5608       4894.0      0.9      0.0          if self.previous_name is None:\n",
      "    36         1          0.0      0.0      0.0              self.previous_name = line[0]\n",
      "    37      5608       4865.0      0.9      0.0          self.current_name = line[0]\n",
      "    38      5608       4160.0      0.7      0.0          if self.current_name != self.previous_name:\n",
      "    39         3       9229.0   3076.3      0.0              self.__reset()\n",
      "    40      5608       4320.0      0.8      0.0          if cols[2] == 0:\n",
      "    41                                                       # the two numbers are bhat and p-value, we need to compute se(bhat)\n",
      "    42                                                       self.data['buffer']['data'].append([line[cols[0]-1], str(get_se(float(line[cols[0]-1]), float(line[cols[1]-1]))), line[cols[1]-1]])\n",
      "    43                                                   else:\n",
      "    44      5608       8565.0      1.5      0.0              self.data['buffer']['data'].append([line[cols[0]-1], line[cols[1]-1], line[cols[2]-1]])\n",
      "    45      5608      26557.0      4.7      0.1          self.data['buffer']['rownames'].append(self.__format_variant_id(line[1]))\n",
      "    46      5608       3621.0      0.6      0.0          return 0\n",
      "\n",
      "Total time: 33.5308 s\n",
      "File: test.py\n",
      "Function: test at line 68\n",
      "\n",
      "Line #      Hits         Time  Per Hit   % Time  Line Contents\n",
      "==============================================================\n",
      "    68                                           @profile\n",
      "    69                                           def test(sumstat_file, gene_list, keep_ensg_version = 0):\n",
      "    70         1          5.0      5.0      0.0      import warnings\n",
      "    71         1       1835.0   1835.0      0.0      import gzip, os\n",
      "    72         1      16813.0  16813.0      0.1      gene_names = [x.strip() for x in open(gene_list).readlines() if x.strip()]\n",
      "    73         1          2.0      2.0      0.0      if keep_ensg_version == 0 and gene_names is not None:\n",
      "    74         1      90417.0  90417.0      0.3          gene_names = [os.path.splitext(x)[0] for x in gene_names]\n",
      "    75                                           \n",
      "    76         1         10.0     10.0      0.0      ssp = SSData(header = True)\n",
      "    77         1          2.0      2.0      0.0      group_counts = 0\n",
      "    78         1         99.0     99.0      0.0      with gzip.open(sumstat_file) as f:\n",
      "    79         1          2.0      2.0      0.0          while True:\n",
      "    80    100001     499124.0      5.0      1.5              line = f.readline()\n",
      "    81    100001   32750353.0    327.5     97.7              quit = ssp.parse(line, keep_ensg_version, include = gene_names)\n",
      "    82    100001      89054.0      0.9      0.3              if quit == -1:\n",
      "    83     94391      73004.0      0.8      0.2                  continue\n",
      "    84      5610       5480.0      1.0      0.0              if ssp.current_name != ssp.previous_name:\n",
      "    85         4          4.0      1.0      0.0                  group_counts += 1\n",
      "    86         4          6.0      1.5      0.0                  ssp.previous_name = ssp.current_name\n",
      "    87      5610       4553.0      0.8      0.0              if quit:\n",
      "    88         1         30.0     30.0      0.0                  break\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "python -m line_profiler test.py.lprof"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From results above we see `ssp.parse` line is 97% the computation, but as suspected, 96.5% computation were spent on checking `line[0] not in include`; that is the line alone is 94% computational time."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now according to [this post](https://stackoverflow.com/questions/7571635/fastest-way-to-check-if-a-value-exist-in-a-list) it seems I should use `set` or `bisect`. Here is the new code using `set`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overwriting test2.py\n"
     ]
    }
   ],
   "source": [
    "%%writefile test2.py\n",
    "cols = [7,6,8]\n",
    "import numpy as np\n",
    "import copy, os\n",
    "\n",
    "class SSData:\n",
    "    def __init__(self, header = False):\n",
    "        self.data = {'buffer':{'data':[], 'rownames':[]}, 'output':{}}\n",
    "        self.header = header\n",
    "        self.previous_name = self.current_name = None\n",
    "        self.ml = max([cols[0], cols[1], cols[2]])\n",
    "\n",
    "    @profile\n",
    "    def parse(self, line, ensg_version = 0, include = set()):\n",
    "        # input line is snp, gene, beta, t, pval and optionally se(bhat)\n",
    "        if not line:\n",
    "            self.__reset()\n",
    "            self.current_name = None\n",
    "            return 1\n",
    "        if isinstance(line, bytes):\n",
    "            line = line.decode()\n",
    "        line = line.split()\n",
    "        if self.header:\n",
    "            self.header = False\n",
    "            return 0\n",
    "        #\n",
    "        line[0] = line[0].strip('\"')\n",
    "        if not line[0] in include:\n",
    "            return -1\n",
    "        if ensg_version == 0:\n",
    "            line[0] = line[0].rsplit('.',1)[0]\n",
    "        #\n",
    "\n",
    "        # the line is not long enough ... there is format issues\n",
    "        if len(line) < self.ml:\n",
    "            return -1\n",
    "        #\n",
    "        if self.previous_name is None:\n",
    "            self.previous_name = line[0]\n",
    "        self.current_name = line[0]\n",
    "        if self.current_name != self.previous_name:\n",
    "            self.__reset()\n",
    "        if cols[2] == 0:\n",
    "            # the two numbers are bhat and p-value, we need to compute se(bhat)\n",
    "            self.data['buffer']['data'].append([line[cols[0]-1], str(get_se(float(line[cols[0]-1]), float(line[cols[1]-1]))), line[cols[1]-1]])\n",
    "        else:\n",
    "            self.data['buffer']['data'].append([line[cols[0]-1], line[cols[1]-1], line[cols[2]-1]])\n",
    "        self.data['buffer']['rownames'].append(self.__format_variant_id(line[1]))\n",
    "        return 0\n",
    "\n",
    "    @staticmethod\n",
    "    def __format_variant_id(value):\n",
    "        value = value.strip('\"').lstrip('chr').split('_')\n",
    "        if len(value) > 4:\n",
    "            # keep it chr, pos, ref, alt\n",
    "            value = value[:4]\n",
    "        return '_'.join(value)\n",
    "\n",
    "    def __reset(self):\n",
    "        self.data['buffer']['data'] = np.array(self.data['buffer']['data'], dtype = float)\n",
    "        self.data['buffer']['rownames'] = np.array(self.data['buffer']['rownames'])\n",
    "        self.data['buffer']['colnames'] = np.array(['beta','se','pval'])\n",
    "        self.data['output'] = copy.deepcopy(self.data['buffer'])\n",
    "        self.data['buffer'] = {'data':[], 'rownames':[]}\n",
    "\n",
    "    def dump(self):\n",
    "        return self.data['output']\n",
    "\n",
    "@profile\n",
    "def test(sumstat_file, gene_list, keep_ensg_version = 0):\n",
    "    import warnings\n",
    "    import gzip\n",
    "    gene_names = set([x.strip() for x in open(gene_list).readlines() if x.strip()])\n",
    "    ssp = SSData(header = True)\n",
    "    group_counts = 0\n",
    "    with gzip.open(sumstat_file) as f:\n",
    "        while True:\n",
    "            line = f.readline()\n",
    "            quit = ssp.parse(line, keep_ensg_version, include = gene_names)\n",
    "            if quit == -1:\n",
    "                continue\n",
    "            if ssp.current_name != ssp.previous_name:\n",
    "                group_counts += 1\n",
    "                ssp.previous_name = ssp.current_name\n",
    "            if quit:\n",
    "                break\n",
    "test('test.txt.gz', 'gtex_v8_sqtl_eur_only.genes_list')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wrote profile results to test2.py.lprof\n",
      "Timer unit: 1e-06 s\n",
      "\n",
      "Total time: 0.721887 s\n",
      "File: test2.py\n",
      "Function: parse at line 12\n",
      "\n",
      "Line #      Hits         Time  Per Hit   % Time  Line Contents\n",
      "==============================================================\n",
      "    12                                               @profile\n",
      "    13                                               def parse(self, line, ensg_version = 0, include = set()):\n",
      "    14                                                   # input line is snp, gene, beta, t, pval and optionally se(bhat)\n",
      "    15    100001      63919.0      0.6      8.9          if not line:\n",
      "    16         1       2636.0   2636.0      0.4              self.__reset()\n",
      "    17         1          2.0      2.0      0.0              self.current_name = None\n",
      "    18         1          1.0      1.0      0.0              return 1\n",
      "    19    100000      80667.0      0.8     11.2          if isinstance(line, bytes):\n",
      "    20    100000      87354.0      0.9     12.1              line = line.decode()\n",
      "    21    100000     122029.0      1.2     16.9          line = line.split()\n",
      "    22    100000      67378.0      0.7      9.3          if self.header:\n",
      "    23         1          1.0      1.0      0.0              self.header = False\n",
      "    24         1          1.0      1.0      0.0              return 0\n",
      "    25                                                   #\n",
      "    26     99999      98355.0      1.0     13.6          line[0] = line[0].strip('\"')\n",
      "    27     99999      73386.0      0.7     10.2          if not line[0] in include:\n",
      "    28     94391      55688.0      0.6      7.7              return -1\n",
      "    29      5608       3706.0      0.7      0.5          if ensg_version == 0:\n",
      "    30      5608       5900.0      1.1      0.8              line[0] = line[0].rsplit('.',1)[0]\n",
      "    31                                                   #\n",
      "    32                                           \n",
      "    33                                                   # the line is not long enough ... there is format issues\n",
      "    34      5608       4607.0      0.8      0.6          if len(line) < self.ml:\n",
      "    35                                                       return -1\n",
      "    36                                                   #\n",
      "    37      5608       3825.0      0.7      0.5          if self.previous_name is None:\n",
      "    38         1          1.0      1.0      0.0              self.previous_name = line[0]\n",
      "    39      5608       4254.0      0.8      0.6          self.current_name = line[0]\n",
      "    40      5608       4406.0      0.8      0.6          if self.current_name != self.previous_name:\n",
      "    41         3       8749.0   2916.3      1.2              self.__reset()\n",
      "    42      5608       4105.0      0.7      0.6          if cols[2] == 0:\n",
      "    43                                                       # the two numbers are bhat and p-value, we need to compute se(bhat)\n",
      "    44                                                       self.data['buffer']['data'].append([line[cols[0]-1], str(get_se(float(line[cols[0]-1]), float(line[cols[1]-1]))), line[cols[1]-1]])\n",
      "    45                                                   else:\n",
      "    46      5608       7613.0      1.4      1.1              self.data['buffer']['data'].append([line[cols[0]-1], line[cols[1]-1], line[cols[2]-1]])\n",
      "    47      5608      19907.0      3.5      2.8          self.data['buffer']['rownames'].append(self.__format_variant_id(line[1]))\n",
      "    48      5608       3397.0      0.6      0.5          return 0\n",
      "\n",
      "Total time: 2.19414 s\n",
      "File: test2.py\n",
      "Function: test at line 68\n",
      "\n",
      "Line #      Hits         Time  Per Hit   % Time  Line Contents\n",
      "==============================================================\n",
      "    68                                           @profile\n",
      "    69                                           def test(sumstat_file, gene_list, keep_ensg_version = 0):\n",
      "    70         1          4.0      4.0      0.0      import warnings\n",
      "    71         1       1387.0   1387.0      0.1      import gzip\n",
      "    72         1      18989.0  18989.0      0.9      gene_names = set([x.strip() for x in open(gene_list).readlines() if x.strip()])\n",
      "    73         1         15.0     15.0      0.0      ssp = SSData(header = True)\n",
      "    74         1          1.0      1.0      0.0      group_counts = 0\n",
      "    75         1         93.0     93.0      0.0      with gzip.open(sumstat_file) as f:\n",
      "    76         1          2.0      2.0      0.0          while True:\n",
      "    77    100001     403364.0      4.0     18.4              line = f.readline()\n",
      "    78    100001    1622132.0     16.2     73.9              quit = ssp.parse(line, keep_ensg_version, include = gene_names)\n",
      "    79    100001      77187.0      0.8      3.5              if quit == -1:\n",
      "    80     94391      62053.0      0.7      2.8                  continue\n",
      "    81      5610       4992.0      0.9      0.2              if ssp.current_name != ssp.previous_name:\n",
      "    82         4          2.0      0.5      0.0                  group_counts += 1\n",
      "    83         4          4.0      1.0      0.0                  ssp.previous_name = ssp.current_name\n",
      "    84      5610       3894.0      0.7      0.2              if quit:\n",
      "    85         1         22.0     22.0      0.0                  break\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "kernprof -l test2.py\n",
    "python -m line_profiler test2.py.lprof"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now that line is 73.9% * 10.2% = 7% computation. Total time to process the test file reduces from 30 sec to 2 sec. \n",
    "\n",
    "**Update** based on the profile i tried to make some small changes in some other places above but there is nothing obvious left to improve. Below shows the difference before and after:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "diff test.py test2.py"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
