var tipuesearch = {"pages": [
{"mainTitle": "Analysis", "title": "Analysis", "text": "Analysis ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis.html#Analysis"},
{"mainTitle": "Analysis", "title": "2017/12", "text": "2017 12 ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis.html#2017/12"},
{"mainTitle": "Analysis", "title": "2017/10", "text": "2017 10 ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis.html#2017/10"},
{"mainTitle": "Analysis", "title": "2017/09", "text": "2017 09 ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis.html#2017/09"},
{"mainTitle": "Analysis", "title": "2017/08", "text": "2017 08 ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis.html#2017/08"},
{"mainTitle": "Analysis", "title": "2017/06", "text": "2017 06 ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis.html#2017/06"},
{"mainTitle": "Analysis", "title": "2017/05", "text": "2017 05 ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis.html#2017/05"},
{"mainTitle": "Analysis", "title": "2017/04", "text": "2017 04 ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis.html#2017/04"},
{"mainTitle": "Prototype", "title": "Prototype", "text": "Prototype ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/prototype.html", "url": "http://stephenslab.github.io/gtex-eqtls/prototype.html#Prototype"},
{"mainTitle": "Prototype", "title": "2017/08", "text": "2017 08 ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/prototype.html", "url": "http://stephenslab.github.io/gtex-eqtls/prototype.html#2017/08"},
{"mainTitle": "Writeup", "title": "Writeup", "text": "Writeup ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/writeup.html", "url": "http://stephenslab.github.io/gtex-eqtls/writeup.html#Writeup"},
{"mainTitle": "Welcome to GTEx eQTL!", "title": "Welcome to GTEx eQTL!", "text": "Welcome to GTEx eQTL ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/index.html", "url": "http://stephenslab.github.io/gtex-eqtls/index.html#Welcome-to-GTEx-eQTL!"},
{"mainTitle": "MASH analysis for Urbut 2017", "title": "MASH analysis for Urbut 2017", "text": "MASH analysis for Urbut 2017 ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html#MASH-analysis-for-Urbut-2017"},
{"mainTitle": "MASH analysis for Urbut 2017", "title": "Covariance pattern discovery", "text": "Covariance pattern discovery ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html#Covariance-pattern-discovery"},
{"mainTitle": "MASH analysis for Urbut 2017", "title": "SFA", "text": "SFA ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html#SFA"},
{"mainTitle": "MASH analysis for Urbut 2017", "title": "Create and refine multi-rank covariance matrices", "text": "Create and refine multi-rank covariance matrices ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html#Create-and-refine-multi-rank-covariance-matrices"},
{"mainTitle": "MASH analysis for Urbut 2017", "title": "Add in single-rank covariance matrices", "text": "Add in single-rank covariance matrices ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html#Add-in-single-rank-covariance-matrices"},
{"mainTitle": "MASH analysis for Urbut 2017", "title": "Fit MASH mixture model", "text": "Fit MASH mixture model ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html#Fit-MASH-mixture-model"},
{"mainTitle": "MASH analysis for Urbut 2017", "title": "Posterior inference", "text": "Posterior inference ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html#Posterior-inference"},
{"mainTitle": "MASH analysis for Urbut 2017", "title": "Run this notebook", "text": "Run this notebook ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Paper.html#Run-this-notebook"},
{"mainTitle": "Compute pairwise LD for selected SNPs", "title": "Compute pairwise LD for selected SNPs", "text": "Compute pairwise LD for selected SNPs ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170828_Compute_LD.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170828_Compute_LD.html#Compute-pairwise-LD-for-selected-SNPs"},
{"mainTitle": "Compute pairwise LD for selected SNPs", "title": "Prepare SNP ID list to extract", "text": "Prepare SNP ID list to extract ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170828_Compute_LD.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170828_Compute_LD.html#Prepare-SNP-ID-list-to-extract"},
{"mainTitle": "Compute pairwise LD for selected SNPs", "title": "From text file", "text": "From text file ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170828_Compute_LD.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170828_Compute_LD.html#From-text-file"},
{"mainTitle": "Compute pairwise LD for selected SNPs", "title": "From HDF5 table row names", "text": "From HDF5 table row names ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170828_Compute_LD.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170828_Compute_LD.html#From-HDF5-table-row-names"},
{"mainTitle": "Compute pairwise LD for selected SNPs", "title": "Compute LD for given list of variants", "text": "Compute LD for given list of variants ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170828_Compute_LD.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170828_Compute_LD.html#Compute-LD-for-given-list-of-variants"},
{"mainTitle": "Compute pairwise LD for selected SNPs", "title": "Examine LD strengths per chromosome", "text": "Examine LD strengths per chromosome ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170828_Compute_LD.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170828_Compute_LD.html#Examine-LD-strengths-per-chromosome"},
{"mainTitle": "Compute pairwise LD for selected SNPs", "title": "Session Info", "text": "Session Info ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170828_Compute_LD.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170828_Compute_LD.html"},
{"mainTitle": "Preparing input data for MASH analysis", "title": "Preparing input data for MASH analysis", "text": "Preparing input data for MASH analysis ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Input_Preparation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Input_Preparation.html#Preparing-input-data-for-MASH-analysis"},
{"mainTitle": "Preparing input data for MASH analysis", "title": "Obtain data", "text": "Obtain data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Input_Preparation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Input_Preparation.html#Obtain-data"},
{"mainTitle": "Preparing input data for MASH analysis", "title": "Extract training and testing data", "text": "Extract training and testing data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Input_Preparation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Input_Preparation.html#Extract-training-and-testing-data"},
{"mainTitle": "Preparing input data for MASH analysis", "title": "Extract from selected list", "text": "Extract from selected list ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Input_Preparation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170829_MASH_Input_Preparation.html#Extract-from-selected-list"},
{"mainTitle": "MASH results on V6 data", "title": "MASH results on V6 data", "text": "MASH results on V6 data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171004_MASH_Results_V6_XX.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171004_MASH_Results_V6_XX.html#MASH-results-on-V6-data"},
{"mainTitle": "MASH results on V6 data", "title": "MASH model fit", "text": "MASH model fit ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171004_MASH_Results_V6_XX.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171004_MASH_Results_V6_XX.html#MASH-model-fit"},
{"mainTitle": "Analyzing a toy example with varbvs and mr-ash", "title": "Analyzing a toy example with varbvs and mr-ash", "text": "Analyzing a toy example with varbvs and mr-ash ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html#Analyzing-a-toy-example-with-varbvs-and-mr-ash"},
{"mainTitle": "Analyzing a toy example with varbvs and mr-ash", "title": "Utility functions", "text": "Utility functions ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html#Utility-functions"},
{"mainTitle": "Analyzing a toy example with varbvs and mr-ash", "title": "Core analysis", "text": "Core analysis ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html#Core-analysis"},
{"mainTitle": "Analyzing a toy example with varbvs and mr-ash", "title": "Check top hits with lasso init", "text": "Check top hits with lasso init ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html#Check-top-hits-with-lasso-init"},
{"mainTitle": "Analyzing a toy example with varbvs and mr-ash", "title": "Credible sets", "text": "Credible sets ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html#Credible-sets"},
{"mainTitle": "Analyzing a toy example with varbvs and mr-ash", "title": "Narrow X to 100 SNPs", "text": "Narrow X to 100 SNPs ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html#Narrow-X-to-100-SNPs"},
{"mainTitle": "Analyzing a toy example with varbvs and mr-ash", "title": "Compare ~7.5K vs ~100 SNPs", "text": "Compare 7.5K vs 100 SNPs ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html#Compare-~7.5K-vs-~100-SNPs"},
{"mainTitle": "Analyzing a toy example with varbvs and mr-ash", "title": "varbvs", "text": "varbvs ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html#varbvs"},
{"mainTitle": "Analyzing a toy example with varbvs and mr-ash", "title": "varbvsmix", "text": "varbvsmix ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html#varbvsmix"},
{"mainTitle": "Analyzing a toy example with varbvs and mr-ash", "title": "The top 4", "text": "The top 4 ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171027_BVS_MR_ASH.html#The-top-4"},
{"mainTitle": "MASH results on V8 data", "title": "MASH results on V8 data", "text": "MASH results on V8 data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171003_MASH_Results.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171003_MASH_Results.html#MASH-results-on-V8-data"},
{"mainTitle": "MASH results on V8 data", "title": "MASH model fit", "text": "MASH model fit ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171003_MASH_Results.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171003_MASH_Results.html#MASH-model-fit"},
{"mainTitle": "MASH results on V8 data", "title": "Posterior quantities", "text": "Posterior quantities ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171003_MASH_Results.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171003_MASH_Results.html#Posterior-quantities"},
{"mainTitle": "MASH results on V8 data", "title": "eQTLs discovered", "text": "eQTLs discovered ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171003_MASH_Results.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171003_MASH_Results.html#eQTLs-discovered"},
{"mainTitle": "MASH results on V8 data", "title": "Data-driven patterns of sharing", "text": "Data-driven patterns of sharing ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171003_MASH_Results.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171003_MASH_Results.html#Data-driven-patterns-of-sharing"},
{"mainTitle": "BIMBAM analysis with selected set of FMO2 SNPs for Thyroid", "title": "BIMBAM analysis with selected set of FMO2 SNPs for Thyroid", "text": "BIMBAM analysis with selected set of FMO2 SNPs for Thyroid ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html#BIMBAM-analysis-with-selected-set-of-FMO2-SNPs-for-Thyroid"},
{"mainTitle": "BIMBAM analysis with selected set of FMO2 SNPs for Thyroid", "title": "Get subset of SNPs", "text": "Get subset of SNPs ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html#Get-subset-of-SNPs"},
{"mainTitle": "BIMBAM analysis with selected set of FMO2 SNPs for Thyroid", "title": "Convert to BIMBAM genotype", "text": "Convert to BIMBAM genotype ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html#Convert-to-BIMBAM-genotype"},
{"mainTitle": "BIMBAM analysis with selected set of FMO2 SNPs for Thyroid", "title": "Prepare BIMBAM phenotype", "text": "Prepare BIMBAM phenotype ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html#Prepare-BIMBAM-phenotype"},
{"mainTitle": "BIMBAM analysis with selected set of FMO2 SNPs for Thyroid", "title": "Run BIMBAM", "text": "Run BIMBAM ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html#Run-BIMBAM"},
{"mainTitle": "BIMBAM analysis with selected set of FMO2 SNPs for Thyroid", "title": "Summarize BIMBAM results", "text": "Summarize BIMBAM results ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html#Summarize-BIMBAM-results"},
{"mainTitle": "BIMBAM analysis with selected set of FMO2 SNPs for Thyroid", "title": "For varbvsmix", "text": "For varbvsmix ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html#For-varbvsmix"},
{"mainTitle": "BIMBAM analysis with selected set of FMO2 SNPs for Thyroid", "title": "For varbvs", "text": "For varbvs ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM.html#For-varbvs"},
{"mainTitle": "Association mapping covariates", "title": "Association mapping covariates", "text": "Association mapping covariates ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170605_Prepare_Covariates.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170605_Prepare_Covariates.html#Association-mapping-covariates"},
{"mainTitle": "Association mapping covariates", "title": "About orthonormal basis", "text": "About orthonormal basis ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170605_Prepare_Covariates.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170605_Prepare_Covariates.html#About-orthonormal-basis"},
{"mainTitle": "MASH results on V6 data", "title": "MASH results on V6 data", "text": "MASH results on V6 data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171006_MASH_Results_V6.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171006_MASH_Results_V6.html#MASH-results-on-V6-data"},
{"mainTitle": "MASH results on V6 data", "title": "MASH model fit", "text": "MASH model fit ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171006_MASH_Results_V6.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171006_MASH_Results_V6.html#MASH-model-fit"},
{"mainTitle": "MASH analysis for GTEx V8 data", "title": "MASH analysis for GTEx V8 data", "text": "MASH analysis for GTEx V8 data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MASH_V8.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MASH_V8.html#MASH-analysis-for-GTEx-V8-data"},
{"mainTitle": "MASH analysis for GTEx V8 data", "title": "Covariance pattern discovery", "text": "Covariance pattern discovery ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MASH_V8.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MASH_V8.html#Covariance-pattern-discovery"},
{"mainTitle": "MASH analysis for GTEx V8 data", "title": "SFA", "text": "SFA ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MASH_V8.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MASH_V8.html#SFA"},
{"mainTitle": "MASH analysis for GTEx V8 data", "title": "Create and refine multi-rank covariance matrices", "text": "Create and refine multi-rank covariance matrices ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MASH_V8.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MASH_V8.html#Create-and-refine-multi-rank-covariance-matrices"},
{"mainTitle": "MASH analysis for GTEx V8 data", "title": "Fit MASH mixture model", "text": "Fit MASH mixture model ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MASH_V8.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MASH_V8.html#Fit-MASH-mixture-model"},
{"mainTitle": "MASH analysis for GTEx V8 data", "title": "Posterior inference", "text": "Posterior inference ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MASH_V8.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MASH_V8.html#Posterior-inference"},
{"mainTitle": "MASH analysis for GTEx V8 data", "title": "Run this notebook", "text": "Run this notebook ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MASH_V8.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MASH_V8.html#Run-this-notebook"},
{"mainTitle": "MASH results on V8 data", "title": "MASH results on V8 data", "text": "MASH results on V8 data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171004_MASH_Results.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171004_MASH_Results.html#MASH-results-on-V8-data"},
{"mainTitle": "MASH results on V8 data", "title": "MASH model fit", "text": "MASH model fit ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171004_MASH_Results.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171004_MASH_Results.html#MASH-model-fit"},
{"mainTitle": "MR-ASH analysis on Lung", "title": "MR-ASH analysis on Lung", "text": "MR-ASH analysis on Lung ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170824_MR_ASH_Lung.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170824_MR_ASH_Lung.html#MR-ASH-analysis-on-Lung"},
{"mainTitle": "GTEx V8 expression and covariate data", "title": "GTEx V8 expression and covariate data", "text": "GTEx V8 expression and covariate data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Expression_Covariate.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Expression_Covariate.html#GTEx-V8-expression-and-covariate-data"},
{"mainTitle": "GTEx V8 expression and covariate data", "title": "Covariates format and computation", "text": "Covariates format and computation ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Expression_Covariate.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Expression_Covariate.html#Covariates-format-and-computation"},
{"mainTitle": "GTEx V8 expression and covariate data", "title": "Expression data format", "text": "Expression data format ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Expression_Covariate.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Expression_Covariate.html#Expression-data-format"},
{"mainTitle": "Extract per-gene per tissue data", "title": "Extract per-gene per tissue data", "text": "Extract per-gene per tissue data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171017_Extract_Full_Tissue_Data.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171017_Extract_Full_Tissue_Data.html#Extract-per-gene-per-tissue-data"},
{"mainTitle": "Converting tissue specific eQTL summary statistics to HDF5", "title": "Converting tissue specific eQTL summary statistics to HDF5", "text": "Converting tissue specific eQTL summary statistics to HDF5 ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html#Converting-tissue-specific-eQTL-summary-statistics-to-HDF5"},
{"mainTitle": "Converting tissue specific eQTL summary statistics to HDF5", "title": "Input data", "text": "Input data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html#Input-data"},
{"mainTitle": "Converting tissue specific eQTL summary statistics to HDF5", "title": "Utility codes", "text": "Utility codes ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html#Utility-codes"},
{"mainTitle": "Converting tissue specific eQTL summary statistics to HDF5", "title": "Convert to HDF5 format", "text": "Convert to HDF5 format ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html#Convert-to-HDF5-format"},
{"mainTitle": "Converting tissue specific eQTL summary statistics to HDF5", "title": "Per tissue conversion", "text": "Per tissue conversion ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html#Per-tissue-conversion"},
{"mainTitle": "Converting tissue specific eQTL summary statistics to HDF5", "title": "Merge per tissue HDF5 to one HDF5", "text": "Merge per tissue HDF5 to one HDF5 ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html#Merge-per-tissue-HDF5-to-one-HDF5"},
{"mainTitle": "Converting tissue specific eQTL summary statistics to HDF5", "title": "Extract data for mash analysis", "text": "Extract data for mash analysis ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html#Extract-data-for-mash-analysis"},
{"mainTitle": "Converting tissue specific eQTL summary statistics to HDF5", "title": "Run workflow", "text": "Run workflow ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170926_eQTLSummary_HDF5.html#Run-workflow"},
{"mainTitle": "RNA-Seq data preprocessing", "title": "RNA-Seq data preprocessing", "text": "RNA-Seq data preprocessing ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170515_RNASeq_Analysis.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170515_RNASeq_Analysis.html#RNA-Seq-data-preprocessing"},
{"mainTitle": "RNA-Seq data preprocessing", "title": "Data file format conversion", "text": "Data file format conversion ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170515_RNASeq_Analysis.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170515_RNASeq_Analysis.html#Data-file-format-conversion"},
{"mainTitle": "RNA-Seq data preprocessing", "title": "Data file format conversion & normalization", "text": "Data file format conversion normalization ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170515_RNASeq_Analysis.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170515_RNASeq_Analysis.html#Data-file-format-conversion-&-normalization"},
{"mainTitle": "RNA-Seq data preprocessing", "title": "A glance at normalized expression data", "text": "A glance at normalized expression data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170515_RNASeq_Analysis.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170515_RNASeq_Analysis.html#A-glance-at-normalized-expression-data"},
{"mainTitle": "RNA-Seq data preprocessing", "title": "PEER factor analysis", "text": "PEER factor analysis ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170515_RNASeq_Analysis.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170515_RNASeq_Analysis.html#PEER-factor-analysis"},
{"mainTitle": "RNA-Seq data preprocessing", "title": "Execute workflows", "text": "Execute workflows ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170515_RNASeq_Analysis.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170515_RNASeq_Analysis.html#Execute-workflows"},
{"mainTitle": "BIMBAM plots using Matthew's code 2007", "title": "BIMBAM plots using Matthew's code 2007", "text": "BIMBAM plots using Matthew's code 2007 ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM_Plots.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_BIMBAM_Plots.html#BIMBAM-plots-using-Matthew's-code-2007"},
{"mainTitle": "Download data from dbGaP", "title": "Download data from dbGaP", "text": "Download data from dbGaP ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170424_dbGaP_Download.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170424_dbGaP_Download.html#Download-data-from-dbGaP"},
{"mainTitle": "Download data from dbGaP", "title": "dbGaP access", "text": "dbGaP access ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170424_dbGaP_Download.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170424_dbGaP_Download.html#dbGaP-access"},
{"mainTitle": "Download data from dbGaP", "title": "GTEx data", "text": "GTEx data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170424_dbGaP_Download.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170424_dbGaP_Download.html#GTEx-data"},
{"mainTitle": "Download data from dbGaP", "title": "Download and decypt the data", "text": "Download and decypt the data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170424_dbGaP_Download.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170424_dbGaP_Download.html#Download-and-decypt-the-data"},
{"mainTitle": "Extract dataset for given genes", "title": "Extract dataset for given genes", "text": "Extract dataset for given genes ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170627_Subset_Genes.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170627_Subset_Genes.html#Extract-dataset-for-given-genes"},
{"mainTitle": "GTEx V8 genotypes", "title": "GTEx V8 genotypes", "text": "GTEx V8 genotypes ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Genotype.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Genotype.html#GTEx-V8-genotypes"},
{"mainTitle": "GTEx V8 genotypes", "title": "Convert data to PLINK format", "text": "Convert data to PLINK format ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Genotype.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Genotype.html#Convert-data-to-PLINK-format"},
{"mainTitle": "GTEx V8 genotypes", "title": "Create cisSNP genotype data files", "text": "Create cisSNP genotype data files ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Genotype.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Genotype.html#Create-cisSNP-genotype-data-files"},
{"mainTitle": "GTEx V8 genotypes", "title": "Annotate genes", "text": "Annotate genes ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Genotype.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Genotype.html#Annotate-genes"},
{"mainTitle": "GTEx V8 genotypes", "title": "Make cis-SNP genotype tables", "text": "Make cis-SNP genotype tables ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Genotype.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170929_Genotype.html#Make-cis-SNP-genotype-tables"},
{"mainTitle": "GTEx V8 genotype data imputation", "title": "GTEx V8 genotype data imputation", "text": "GTEx V8 genotype data imputation ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171023_Imputation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171023_Imputation.html#GTEx-V8-genotype-data-imputation"},
{"mainTitle": "GTEx V8 genotype data imputation", "title": "Genotype missingness summary", "text": "Genotype missingness summary ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171023_Imputation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171023_Imputation.html#Genotype-missingness-summary"},
{"mainTitle": "GTEx V8 genotype data imputation", "title": "Data processing for imputation server", "text": "Data processing for imputation server ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171023_Imputation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171023_Imputation.html#Data-processing-for-imputation-server"},
{"mainTitle": "GTEx V8 genotype data imputation", "title": "Imputation", "text": "Imputation ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171023_Imputation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171023_Imputation.html#Imputation"},
{"mainTitle": "GTEx V8 genotype data imputation", "title": "Configuration", "text": "Configuration ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171023_Imputation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171023_Imputation.html#Configuration"},
{"mainTitle": "GTEx V8 genotype data imputation", "title": "Outcome", "text": "Outcome ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171023_Imputation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171023_Imputation.html#Outcome"},
{"mainTitle": "GTEx V8 genotype data imputation", "title": "Imputation result summary", "text": "Imputation result summary ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171023_Imputation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171023_Imputation.html#Imputation-result-summary"},
{"mainTitle": "Imputation data post-processing and PCA analysis", "title": "Imputation data post-processing and PCA analysis", "text": "Imputation data post-processing and PCA analysis ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170523_Genotype_Formatting_PCA.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170523_Genotype_Formatting_PCA.html#Imputation-data-post-processing-and-PCA-analysis"},
{"mainTitle": "Imputation data post-processing and PCA analysis", "title": "Genotype filtering and formatting", "text": "Genotype filtering and formatting ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170523_Genotype_Formatting_PCA.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170523_Genotype_Formatting_PCA.html#Genotype-filtering-and-formatting"},
{"mainTitle": "Imputation data post-processing and PCA analysis", "title": "VCF to PLINK conversion", "text": "VCF to PLINK conversion ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170523_Genotype_Formatting_PCA.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170523_Genotype_Formatting_PCA.html#VCF-to-PLINK-conversion"},
{"mainTitle": "Imputation data post-processing and PCA analysis", "title": "Remove imputed variant sites", "text": "Remove imputed variant sites ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170523_Genotype_Formatting_PCA.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170523_Genotype_Formatting_PCA.html#Remove-imputed-variant-sites"},
{"mainTitle": "Imputation data post-processing and PCA analysis", "title": "Global ancestry PCA / MDS analysis", "text": "Global ancestry PCA MDS analysis ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170523_Genotype_Formatting_PCA.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170523_Genotype_Formatting_PCA.html#Global-ancestry-PCA-/-MDS-analysis"},
{"mainTitle": "Imputation data post-processing and PCA analysis", "title": "Plot Broad PCA results", "text": "Plot Broad PCA results ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170523_Genotype_Formatting_PCA.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170523_Genotype_Formatting_PCA.html#Plot-Broad-PCA-results"},
{"mainTitle": "Imputation data post-processing and PCA analysis", "title": "PCA results", "text": "PCA results ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170523_Genotype_Formatting_PCA.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170523_Genotype_Formatting_PCA.html#PCA-results"},
{"mainTitle": "Processing GTEx V8 expression data for various summary statistics", "title": "Processing GTEx V8 expression data for various summary statistics", "text": "Processing GTEx V8 expression data for various summary statistics ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171212_Median_TPM_GTEx_V8.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171212_Median_TPM_GTEx_V8.html#Processing-GTEx-V8-expression-data-for-various-summary-statistics"},
{"mainTitle": "Extract cis-SNP", "title": "Extract cis-SNP", "text": "Extract cis-SNP ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170530_CisSNP.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170530_CisSNP.html#Extract-cis-SNP"},
{"mainTitle": "Extract cis-SNP", "title": "Annotate genes", "text": "Annotate genes ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170530_CisSNP.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170530_CisSNP.html#Annotate-genes"},
{"mainTitle": "Extract cis-SNP", "title": "Make cis-SNP tables", "text": "Make cis-SNP tables ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170530_CisSNP.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170530_CisSNP.html#Make-cis-SNP-tables"},
{"mainTitle": "MASH results on V6 data", "title": "MASH results on V6 data", "text": "MASH results on V6 data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171004_MASH_Results_V6.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171004_MASH_Results_V6.html#MASH-results-on-V6-data"},
{"mainTitle": "MASH results on V6 data", "title": "MASH model fit", "text": "MASH model fit ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171004_MASH_Results_V6.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171004_MASH_Results_V6.html#MASH-model-fit"},
{"mainTitle": "Compare MASH results before and after without SNPs in LD", "title": "Compare MASH results before and after without SNPs in LD", "text": "Compare MASH results before and after without SNPs in LD ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170830_Compare_MASH_No_LD.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170830_Compare_MASH_No_LD.html#Compare-MASH-results-before-and-after-without-SNPs-in-LD"},
{"mainTitle": "Compare MASH results before and after without SNPs in LD", "title": "Compare lfsr", "text": "Compare lfsr ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170830_Compare_MASH_No_LD.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170830_Compare_MASH_No_LD.html#Compare-lfsr"},
{"mainTitle": "Compare MASH results before and after without SNPs in LD", "title": "Compare sharing by magnitude", "text": "Compare sharing by magnitude ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170830_Compare_MASH_No_LD.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170830_Compare_MASH_No_LD.html#Compare-sharing-by-magnitude"},
{"mainTitle": "GTEx V7 genotype data imputation", "title": "GTEx V7 genotype data imputation", "text": "GTEx V7 genotype data imputation ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170518_Imputation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170518_Imputation.html#GTEx-V7-genotype-data-imputation"},
{"mainTitle": "GTEx V7 genotype data imputation", "title": "Genotype missingness summary", "text": "Genotype missingness summary ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170518_Imputation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170518_Imputation.html#Genotype-missingness-summary"},
{"mainTitle": "GTEx V7 genotype data imputation", "title": "Data processing for imputation server", "text": "Data processing for imputation server ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170518_Imputation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170518_Imputation.html#Data-processing-for-imputation-server"},
{"mainTitle": "GTEx V7 genotype data imputation", "title": "Imputation", "text": "Imputation ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170518_Imputation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170518_Imputation.html#Imputation"},
{"mainTitle": "GTEx V7 genotype data imputation", "title": "Configuration", "text": "Configuration ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170518_Imputation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170518_Imputation.html#Configuration"},
{"mainTitle": "GTEx V7 genotype data imputation", "title": "Outcome", "text": "Outcome ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170518_Imputation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170518_Imputation.html#Outcome"},
{"mainTitle": "GTEx V7 genotype data imputation", "title": "Imputation result summary", "text": "Imputation result summary ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170518_Imputation.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170518_Imputation.html#Imputation-result-summary"},
{"mainTitle": "MR ASH on single tissue data", "title": "MR ASH on single tissue data", "text": "MR ASH on single tissue data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MR_ASH_V8.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MR_ASH_V8.html#MR-ASH-on-single-tissue-data"},
{"mainTitle": "MR ASH on single tissue data", "title": "Run workflow on remote computer", "text": "Run workflow on remote computer ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MR_ASH_V8.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171002_MR_ASH_V8.html#Run-workflow-on-remote-computer"},
{"mainTitle": "MASH results on V8 data", "title": "MASH results on V8 data", "text": "MASH results on V8 data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171005_MASH_Results_V8.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171005_MASH_Results_V8.html#MASH-results-on-V8-data"},
{"mainTitle": "MASH results on V8 data", "title": "MASH model fit", "text": "MASH model fit ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171005_MASH_Results_V8.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171005_MASH_Results_V8.html#MASH-model-fit"},
{"mainTitle": "MR ASH on GTEx genes", "title": "MR ASH on GTEx genes", "text": "MR ASH on GTEx genes ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20170814_MR_ASH_V7.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20170814_MR_ASH_V7.html#MR-ASH-on-GTEx-genes"},
{"mainTitle": "QTL data preprocessing for Yuxin Zou", "title": "QTL data preprocessing for Yuxin Zou", "text": "QTL data preprocessing for Yuxin Zou ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html#QTL-data-preprocessing-for-Yuxin-Zou"},
{"mainTitle": "QTL data preprocessing for Yuxin Zou", "title": "Input data", "text": "Input data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html#Input-data"},
{"mainTitle": "QTL data preprocessing for Yuxin Zou", "title": "Utility codes", "text": "Utility codes ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html#Utility-codes"},
{"mainTitle": "QTL data preprocessing for Yuxin Zou", "title": "Convert to HDF5 format", "text": "Convert to HDF5 format ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html#Convert-to-HDF5-format"},
{"mainTitle": "QTL data preprocessing for Yuxin Zou", "title": "Per study conversion", "text": "Per study conversion ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html#Per-study-conversion"},
{"mainTitle": "QTL data preprocessing for Yuxin Zou", "title": "Merge per tissue HDF5 to one HDF5", "text": "Merge per tissue HDF5 to one HDF5 ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html#Merge-per-tissue-HDF5-to-one-HDF5"},
{"mainTitle": "QTL data preprocessing for Yuxin Zou", "title": "Extract data for mash analysis", "text": "Extract data for mash analysis ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html#Extract-data-for-mash-analysis"},
{"mainTitle": "QTL data preprocessing for Yuxin Zou", "title": "Run workflow", "text": "Run workflow ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171010_eQTLSummary_HDF5.html#Run-workflow"},
{"mainTitle": "Summary of single tissue analysis results", "title": "Summary of single tissue analysis results", "text": "Summary of single tissue analysis results ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_Single_Tissue_MR_ASH_Summary.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_Single_Tissue_MR_ASH_Summary.html#Summary-of-single-tissue-analysis-results"},
{"mainTitle": "Summary of single tissue analysis results", "title": "Number of eQTLs per gene", "text": "Number of eQTLs per gene ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_Single_Tissue_MR_ASH_Summary.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_Single_Tissue_MR_ASH_Summary.html#Number-of-eQTLs-per-gene"},
{"mainTitle": "Summary of single tissue analysis results", "title": "Summary for Thyroid data", "text": "Summary for Thyroid data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_Single_Tissue_MR_ASH_Summary.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_Single_Tissue_MR_ASH_Summary.html#Summary-for-Thyroid-data"},
{"mainTitle": "Summary of single tissue analysis results", "title": "Distribution of eQTLs per gene", "text": "Distribution of eQTLs per gene ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_Single_Tissue_MR_ASH_Summary.html", "url": "http://stephenslab.github.io/gtex-eqtls/analysis/20171030_Single_Tissue_MR_ASH_Summary.html#Distribution-of-eQTLs-per-gene"},
{"mainTitle": "PEER analysis result not orthogonal", "title": "PEER analysis result not orthogonal", "text": "PEER analysis result not orthogonal ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/prototype/20170828_PEER_Not_Orthogonal.html", "url": "http://stephenslab.github.io/gtex-eqtls/prototype/20170828_PEER_Not_Orthogonal.html#PEER-analysis-result-not-orthogonal"},
{"mainTitle": "PEER analysis result not orthogonal", "title": "Orthonormal basis via SVD", "text": "Orthonormal basis via SVD ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/prototype/20170828_PEER_Not_Orthogonal.html", "url": "http://stephenslab.github.io/gtex-eqtls/prototype/20170828_PEER_Not_Orthogonal.html#Orthonormal-basis-via-SVD"},
{"mainTitle": "PEER analysis result not orthogonal", "title": "Orthonormal basis via QR", "text": "Orthonormal basis via QR ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/prototype/20170828_PEER_Not_Orthogonal.html", "url": "http://stephenslab.github.io/gtex-eqtls/prototype/20170828_PEER_Not_Orthogonal.html#Orthonormal-basis-via-QR"},
{"mainTitle": "You need a password to continue", "title": "You need a password to continue", "text": "You need a password to continue ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/writeup/Meetings.html", "url": "http://stephenslab.github.io/gtex-eqtls/writeup/Meetings.html"},
{"mainTitle": "GTEx V7 eQTL data analysis procedure", "title": "GTEx V7 eQTL data analysis procedure", "text": "GTEx V7 eQTL data analysis procedure ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html", "url": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html#GTEx-V7-eQTL-data-analysis-procedure"},
{"mainTitle": "GTEx V7 eQTL data analysis procedure", "title": "Preprocessing", "text": "Preprocessing ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html", "url": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html#Preprocessing"},
{"mainTitle": "GTEx V7 eQTL data analysis procedure", "title": "What we have", "text": "What we have ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html", "url": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html#What-we-have"},
{"mainTitle": "GTEx V7 eQTL data analysis procedure", "title": "What we need to do", "text": "What we need to do ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html", "url": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html#What-we-need-to-do"},
{"mainTitle": "GTEx V7 eQTL data analysis procedure", "title": "Genotype data", "text": "Genotype data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html", "url": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html#Genotype-data"},
{"mainTitle": "GTEx V7 eQTL data analysis procedure", "title": "RNA-Seq data", "text": "RNA-Seq data ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html", "url": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html#RNA-Seq-data"},
{"mainTitle": "GTEx V7 eQTL data analysis procedure", "title": "Covariates", "text": "Covariates ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html", "url": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html#Covariates"},
{"mainTitle": "GTEx V7 eQTL data analysis procedure", "title": "FastQTL analysis", "text": "FastQTL analysis ", "tags": "", "mainUrl": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html", "url": "http://stephenslab.github.io/gtex-eqtls/writeup/GTEx7_Analysis_Plan.html#FastQTL-analysis"}
]};
